{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Using Python  access SSBD OMERO and IDR for searching meta-data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Updated: 2019-04-02"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Loading json library allows iPython to deal with JSON formatted data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "SSBD_OMERO_URL = \"http://ssbd.qbic.riken.jp/image\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "IMAGE_ID=1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Testing some usage of SSBD-OMERO API"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Image link: http://ssbd.qbic.riken.jp/image/webclient/?show=image-1\n",
      "Image viewer link: http://ssbd.qbic.riken.jp/image/webclient/img_detail/1/\n",
      "Thumbnail URL: http://ssbd.qbic.riken.jp/image/webclient/render_thumbnail/1/\n",
      "Attributes URL: http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&image=1\n"
     ]
    }
   ],
   "source": [
    "image_url = \"{base}/webclient/?show=image-{image_id}\"\n",
    "image_viewer = \"{base}/webclient/img_detail/{image_id}/\"\n",
    "thumbnail_url = \"{base}/webclient/render_thumbnail/{image_id}/\"\n",
    "attributes_url = \"{base}/webclient/api/annotations/?type=map&image={image_id}\"\n",
    "\n",
    "print('Image link:', image_url.format(**{'base': SSBD_OMERO_URL, 'image_id': IMAGE_ID}))\n",
    "print('Image viewer link:', image_viewer.format(**{'base': SSBD_OMERO_URL, 'image_id': IMAGE_ID}))\n",
    "print('Thumbnail URL:', thumbnail_url.format(**{'base': SSBD_OMERO_URL, 'image_id': IMAGE_ID}))\n",
    "print('Attributes URL:', attributes_url.format(**{'base': SSBD_OMERO_URL, 'image_id': IMAGE_ID}))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "ANNOTATE_URL = \"{base}/webclient/api/annotations/{image_id}/\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      " {'annotations': [{'description': None, 'class': 'MapAnnotationI', 'date': '2017-09-20T21:51:36Z', 'link': {'owner': {'id': 2}, 'date': '2017-09-20T21:51:36Z', 'id': 36970, 'parent': {'id': 1, 'name': 'img0000.ipm', 'class': 'ImageI'}, 'permissions': {'canAnnotate': False, 'canEdit': False, 'canDelete': False, 'canLink': False}}, 'owner': {'id': 2}, 'values': [['License', 'CC BY-NC-SA'], ['Contactname', 'Shuichi Onami'], ['Organization', 'RIKEN'], ['Department', 'Quantitative Biology Center'], ['Laboratory', 'Laboratory for Developmental Dynamics'], ['Contributers', 'Eru Adachi, Eriko Masuda, Yoko Nagai, Yoko Suzuki, Taeko Oguro, Koji Kyoda, Shuichi Onami'], ['Organism', 'C. elegans'], ['PubMed', 'https://www.ncbi.nlm.nih.gov/pubmed/23172286']], 'ns': 'openmicroscopy.org/omero/client/mapAnnotation', 'id': 37442, 'permissions': {'canAnnotate': False, 'canEdit': False, 'canDelete': False, 'canLink': False}}], 'experimenters': [{'lastName': 'data', 'omeName': 'public_data', 'id': 2, 'firstName': 'public'}]}\n",
      "\n",
      " http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&image=1\n"
     ]
    }
   ],
   "source": [
    "qs = {'base': SSBD_OMERO_URL, 'image_id': IMAGE_ID}\n",
    "\n",
    "url = attributes_url.format(**qs)\n",
    "r = requests.session().get(url)\n",
    "\n",
    "\n",
    "if r.status_code == 200:\n",
    "    metadata = r.json()\n",
    "    print(\"\\n\", metadata)\n",
    "    print(\"\\n\", url)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Summary of all the APIs that will be used"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* Summary of API for SSBD-OMERO\n",
    "        * List all the projects in OMERO\n",
    "                http://ssbd.qbic.riken.jp/image/webgateway/proj/list/\n",
    "        * List the meta data of a project\n",
    "                http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=2\n",
    "        * List all the dataset in a project\n",
    "                http://ssbd.qbic.riken.jp/image/webgateway/proj/1/children/\n",
    "        * List all the images in a dataset\n",
    "                http://ssbd.qbic.riken.jp/image/webgateway/dataset/1/children/\n",
    "        * List the attributes of an image\n",
    "                http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&image=25021\n",
    "        * Image link:\n",
    "                http://ssbd.qbic.riken.jp/image/webclient/?show=image-25021\n",
    "        * Image viewer link:\n",
    "                http://ssbd.qbic.riken.jp/image/webclient/img_detail/25021/\n",
    "        * Thumbnail:\n",
    "                http://ssbd.qbic.riken.jp/image/webclient/render_thumbnail/25021/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Reference links to OMERO APIs\n",
    "\n",
    "https://idr.openmicroscopy.org/about/api.html\n",
    "\n",
    "https://docs.openmicroscopy.org/omero/5.4.0/developers/json-api.html\n",
    "\n",
    "https://docs.openmicroscopy.org/omero/5.4.10/developers/Web/WebGateway.html?highlight=webgateway\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Pilot code to test the concept of combining the search of meta data in both SSBD-OMERO and in IDR.\n",
    "\n",
    "As far as I understand there is no direct API that can search for annotation. \n",
    "\n",
    "In order to do the search, a list of projects is retrieved. By iterating through the list of projects, annotation data are retrieved and a list of key value pairs is created. Search is then done by searching through the key, value pairs. It is possible to extend this method to search for annotations within the datasets. \n",
    "\n",
    "For a server application, the list of projects and attributes can be prefetched once a day or once every update. \n",
    "\n",
    "The main disadvantage of this approach is that it is not scalable when the list of projects and their attributes become large. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## function get_project() returns annotation data as a dictionary given base URL and a project ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_project(base_url,projectid):\n",
    "    qs = {'base': base_url, 'project': projectid}\n",
    "    dataset_attributes_url = \"{base}/webclient/api/annotations/?type=map&project={project}\"\n",
    "    url = dataset_attributes_url.format(**qs)\n",
    "    r = requests.session().get(url)\n",
    "\n",
    "    if r.status_code == 200:\n",
    "        metadata = r.json()\n",
    "        data = {}\n",
    "        for a in metadata['annotations']:\n",
    "            namespace = a['ns']\n",
    "            for v in a['values']:\n",
    "                data[v[0]] = v[1]\n",
    "        return(data)\n",
    "           \n",
    "    else:\n",
    "        print(\"error\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'License': 'CC BY-NC-SA', 'Contactname': 'Shuichi Onami', 'Organization': 'RIKEN', 'Department': 'Quantitative Biology Center', 'Laboratory': 'Laboratory for Developmental Dynamics', 'Contributers': 'Eru Adachi, Eriko Masuda, Yoko Nagai, Yoko Suzuki, Taeko Oguro, Koji Kyoda, Shuichi Onami', 'Organism': 'C. elegans', 'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/23172286'}\n"
     ]
    }
   ],
   "source": [
    "result = get_project(SSBD_OMERO_URL, 1)\n",
    "print(result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'License': 'CC BY-NC-SA',\n",
       " 'Contactname': 'Shuichi Onami',\n",
       " 'Organization': 'RIKEN',\n",
       " 'Department': 'Quantitative Biology Center',\n",
       " 'Laboratory': 'Laboratory for Developmental Dynamics',\n",
       " 'Contributers': 'Eru Adachi, Eriko Masuda, Yoko Nagai, Yoko Suzuki, Taeko Oguro, Koji Kyoda, Shuichi Onami',\n",
       " 'Organism': 'C. elegans',\n",
       " 'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/23172286'}"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "result"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Searching for values on all the keys to match koji"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Eru Adachi, Eriko Masuda, Yoko Nagai, Yoko Suzuki, Taeko Oguro, Koji Kyoda, Shuichi Onami']"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[value for key, value in result.items() if 'koji' in value.lower()]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## function get_all_projects() return a list of projects"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_all_projects(base_url): # http://ssbd.qbic.riken.jp/image/webgateway/proj/list/\n",
    "    qs = {'base': base_url, 'type': 'proj'}\n",
    "    dataset_attributes_url = \"{base}/webgateway/{type}/list\"\n",
    "    url = dataset_attributes_url.format(**qs)\n",
    "    # print(url)\n",
    "    r = requests.session().get(url)\n",
    "    if r.status_code == 200:\n",
    "        metadata = r.json()\n",
    "        project_id_list = []\n",
    "        for a in metadata:\n",
    "            project_id_list.append(a)\n",
    "        return(project_id_list)\n",
    "    else:\n",
    "        print(\"error\")    \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Getting all the projects in SSBD-OMERO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'description': 'See details in Inoue and Kondo (2016) Scientific Reports, 6, 33689.',\n",
       "  'id': 68,\n",
       "  'name': '38-Inoue-ShellStruct'},\n",
       " {'description': 'See details in Takayama and Onami. (2016) Cell Reports, 15(3), 625-637',\n",
       "  'id': 11,\n",
       "  'name': '12-Takayama-Oocyte'},\n",
       " {'description': 'See details in Kyoda et al. (2013) Nucleic Acids Res 41, D732-D737',\n",
       "  'id': 1,\n",
       "  'name': '2-Kyoda-WormEmbryoRNAi'},\n",
       " {'description': 'See details in Azuma and Onami (2017) BMC Bioinformatics, 18(1): 307.',\n",
       "  'id': 101,\n",
       "  'name': '20-Azuma-WormMembrane'},\n",
       " {'description': 'See details in Toyoshima et al. (2016) PLoS Computational Biology, 12(6): e1004970',\n",
       "  'id': 4,\n",
       "  'name': '11-Toyoshima-NeuAct'},\n",
       " {'description': 'See details in Arata et al. (2016) Cell Reports, 16, 1-13.',\n",
       "  'id': 67,\n",
       "  'name': '37-Arata-MolDynPar2'},\n",
       " {'description': 'See details in Matsuda et al. (2012) Sci. Signal., 5, ra31.',\n",
       "  'id': 61,\n",
       "  'name': '31-Matsuda-SigDyn'},\n",
       " {'description': 'See details in Takai et al. (2015) PNAS, 112(14): 4352-4356.',\n",
       "  'id': 71,\n",
       "  'name': '43-Takai-SubcellStructONL'},\n",
       " {'description': 'See details in Hayashi and Okada (2015) Mol Biol Cell, 26(9): 1743-1751.',\n",
       "  'id': 64,\n",
       "  'name': '34-Hayashi-CellDynSDSRM'},\n",
       " {'description': 'See details in Matsuda et al. (2015) Nat. Commun., 6, 6195',\n",
       "  'id': 62,\n",
       "  'name': '32-Matsuda-SigDyn'},\n",
       " {'description': 'See details in Kondo and Hayashi (2013) Nature, 494, 125-129.',\n",
       "  'id': 52,\n",
       "  'name': '22-Kondo-CellDyn'},\n",
       " {'description': 'See details in Yasui et al. (2014) PLoS Computational Biology, 10(9): e1003817.',\n",
       "  'id': 51,\n",
       "  'name': '21-Yasui-MolDynPTEN'},\n",
       " {'description': 'See details in Shibata et al. (2012) J. Cell Sci., 125: 5138-5150.',\n",
       "  'id': 69,\n",
       "  'name': '39-Shibata-MolDynPTEN'},\n",
       " {'description': 'See details in Komatsuzaki et al. (2015) Small, 11(12):1396-1401',\n",
       "  'id': 10,\n",
       "  'name': '10-Komatsuzaki-MolDyn'},\n",
       " {'description': 'See details in Shirai et al. (2016) PNAS, 113(1): E81-E90.',\n",
       "  'id': 53,\n",
       "  'name': '23-Shirai-ESCellTissue'},\n",
       " {'description': 'See details in Torisawa et al. (2016) Biophysical Journal, 111(2): 373-385.',\n",
       "  'id': 66,\n",
       "  'name': '36-Torisawa-MolDynEg5'},\n",
       " {'description': 'See details in Yamao et al. (2015) Scientific Reports, 5: 17527.',\n",
       "  'id': 63,\n",
       "  'name': '33-Yamao-MolDynRho'},\n",
       " {'description': 'See details in Kunida et al. (2012) Journal of Cell Science, 125(10): 2381-2392',\n",
       "  'id': 5,\n",
       "  'name': '18-Kunida-MolDynRho'},\n",
       " {'description': 'See details in Kanemura et al. (2013) Scientific Reports, 3(2334): 2334.',\n",
       "  'id': 54,\n",
       "  'name': '24-Kanemura-iPSCellDyn'},\n",
       " {'description': 'See details in Yonemura (2014) PLoS ONE, 9(11): e112922.',\n",
       "  'id': 65,\n",
       "  'name': '35-Yonemura-CellDyn'},\n",
       " {'description': 'See details in Watabe et al. (2015) PLoS One, 10(7): e0130089',\n",
       "  'id': 2,\n",
       "  'name': '13-Watabe-MolDyn'},\n",
       " {'description': 'See details in Ke et al. (2016) Cell Reports, 14: 2718-2732.',\n",
       "  'id': 59,\n",
       "  'name': '29-Ke-SeeDB2'},\n",
       " {'description': 'See details in Bashar et al. (2012) PLoS ONE 7, e35550.',\n",
       "  'id': 9,\n",
       "  'name': '8-Bashar-MouseEmbryo'},\n",
       " {'description': 'See details in Susaki et al. (2014) Cell, 157(3): 726–739.',\n",
       "  'id': 106,\n",
       "  'name': '48-Susaki-MouseBrainCUBIC'},\n",
       " {'description': 'See details in Susaki et al. (2015) Nature Protocols, 10: 1709–1727.',\n",
       "  'id': 105,\n",
       "  'name': '49-Susaki-MouseBrainCUBIC'},\n",
       " {'description': 'See details in Tainaka et al. (2014) Cell, 159(4): 911–924.',\n",
       "  'id': 104,\n",
       "  'name': '50-Tainaka-MouseBodyCUBIC'},\n",
       " {'description': 'See details in Sakakibara et al. (2015) Nat Commun. 2013; 4:2212.',\n",
       "  'id': 70,\n",
       "  'name': '40-Sakakibara-KTDyn'},\n",
       " {'description': 'See details in Ochiai et al. (2015) Nucleic Acids Research, 43(19): e127',\n",
       "  'id': 12,\n",
       "  'name': '14-Ochiai-MolDynMS2'},\n",
       " {'description': 'See details in Harima et al. (2013) Cell Reports, 3(1): 1-7',\n",
       "  'id': 3,\n",
       "  'name': '15-Harima-GeneExp'},\n",
       " {'description': 'See details in Ke et al. (2013) Nature Neuroscience, 16: 1154-1161.',\n",
       "  'id': 55,\n",
       "  'name': '25-Ke-SeeDB'},\n",
       " {'description': 'See details in Ichimura et al. (2016) Scientific Reports, 6: 37562.',\n",
       "  'id': 103,\n",
       "  'name': '53-Ichimura-CellDyn'},\n",
       " {'description': 'See details in Kinoshita et al. (2016) Retinal Cell Biology, 57(1): 153-161.',\n",
       "  'id': 56,\n",
       "  'name': '26-Kinoshita-iPSCellDyn'},\n",
       " {'description': 'See details in Iwasaki et al. (2016) PLoS ONE, 11(7): e0158282.',\n",
       "  'id': 57,\n",
       "  'name': '27-Iwasaki-iPSCellDyn'},\n",
       " {'description': 'See details in Assawachananont et al. (2014) Stem Cell Reports, 2(5): 662-674.',\n",
       "  'id': 58,\n",
       "  'name': '28-Assawachananont-iPSCellTissue'},\n",
       " {'description': 'See details in Tanaka (2014) Micromachines, 5(2): 289-299',\n",
       "  'id': 8,\n",
       "  'name': '19-Tanaka-FluidDyn'},\n",
       " {'description': 'See details in Aoki et al. (2013) Molecular Cell, 52(4): 529-540',\n",
       "  'id': 6,\n",
       "  'name': '17-Aoki-MolDyn'},\n",
       " {'description': 'See details in Inomata et al. (2013) Cell, 153(6): 1296-1311.',\n",
       "  'id': 60,\n",
       "  'name': '30-Inomata-CellDyn'},\n",
       " {'description': 'See details in Suzuki et al. (2015) Genes to Cells, 21(4): 358-369.',\n",
       "  'id': 102,\n",
       "  'name': '52-Suzuki-FrogTailDyn'},\n",
       " {'description': 'See details in Tanaka and Fujita (2015) Sensors and Actuators B: Chemical, 210(1): 267-272',\n",
       "  'id': 7,\n",
       "  'name': '16-Tanaka-FluidDyn'},\n",
       " {'description': 'See details in Herawati et al. (2016) J. Cell Biol., 214(5): 571-586.',\n",
       "  'id': 152,\n",
       "  'name': '57-Herawati-CellDyn'},\n",
       " {'description': 'See details in Minegishi et al. (2017) Dev Cell, 40(5): 439-452.',\n",
       "  'id': 154,\n",
       "  'name': '75-Minegishi-NodeCell'},\n",
       " {'description': 'See details in Yasugi et al. (2017) Scientific Reports, 7(1): 1619.',\n",
       "  'id': 158,\n",
       "  'name': '81-Yasugi-DmeCellDyn'},\n",
       " {'description': 'See details in Ikeda et al. (2018) Nat Commun, 9(1): 1387.',\n",
       "  'id': 159,\n",
       "  'name': '82-Ikeda-GeneExp'},\n",
       " {'description': 'See details in Takagi et al. (2017) Clinical Ophthalmology, 11: 1219-1225.',\n",
       "  'id': 156,\n",
       "  'name': '78-Takagi-CellDyn'},\n",
       " {'description': 'See details in Noguchi et al. (2015) Cell Reports, 13(12): 2679–2686.',\n",
       "  'id': 171,\n",
       "  'name': '47-Noguchi-CellDyn'},\n",
       " {'description': 'See details in Jin et al. (2016) Med. Chem. Commun., 7(4): 623-631.',\n",
       "  'id': 168,\n",
       "  'name': '106-Jin-BreastTumorFlu'},\n",
       " {'description': 'See details in Tsuboi et al. (2017) Chem. Commun., 53(68): 9450-9453.',\n",
       "  'id': 169,\n",
       "  'name': '107-Tsuboi-BreastTumorFlu'},\n",
       " {'description': 'See details in Makino et al. (2016) FASEB Journal, 31(4): 1301-1322.',\n",
       "  'id': 170,\n",
       "  'name': '110-Makino-MembraneDyn'},\n",
       " {'description': 'See details in Maekawa et al. (2015) Current Eye Research, 41(4): 558-568.',\n",
       "  'id': 151,\n",
       "  'name': '55-Maekawa-ESCellDyn'},\n",
       " {'description': 'See details in Yano et al. (2013) J. Cell Biol., 203(4): 605-614.',\n",
       "  'id': 173,\n",
       "  'name': '59-Yano-MicrotubuleImmuno'},\n",
       " {'description': 'See details in Bansod et al. (2017) Development, 144(17): 3156-3167.',\n",
       "  'id': 153,\n",
       "  'name': '74-Bansod-NSCellHes5'},\n",
       " {'description': 'See details in Aoki et al. (2017) Developmental Cell, 43(3): 305-317.',\n",
       "  'id': 155,\n",
       "  'name': '77-Aoki-CellDynERK'},\n",
       " {'description': 'See details in Inomata et al. (2017) Chem Commun, 53(81): 11245-11248.',\n",
       "  'id': 157,\n",
       "  'name': '79-Inomata-HeLaCell'},\n",
       " {'description': 'See details in Fumoto et al. (2017) Development, 144(1): 151-162.',\n",
       "  'id': 160,\n",
       "  'name': '83-Fumoto-LungDevWnt'},\n",
       " {'description': 'See details in Ito et al. (2017) Nat Commun, 8(1): 1834-1849.',\n",
       "  'id': 161,\n",
       "  'name': '84-Ito-CellDyn'},\n",
       " {'description': 'See details in Kosodo et al. (2017) Biochem Biophys Res Commun, 483(1): 94-100.',\n",
       "  'id': 162,\n",
       "  'name': '85-Kosodo-NeuralballDyn'},\n",
       " {'description': 'See details in Koeberle et al. (2017) Scientific Reports, 7(1): 13409.',\n",
       "  'id': 172,\n",
       "  'name': '87-CorneliaKoeverle-SpineDyn'},\n",
       " {'description': 'See details in Ishiwata et al. (2017) Molecular Crystals and Liquid Crystals, 647(1): 127-150.',\n",
       "  'id': 163,\n",
       "  'name': '90-Ishiwata-SpindleDyn'},\n",
       " {'description': 'See details in Konagaya et al. (2017) Cell Reports, 21(9): 2628-2638.',\n",
       "  'id': 164,\n",
       "  'name': '91-Konagaya-AMPKactivity'},\n",
       " {'description': 'See details in Uda et al. (2017) PNAS, 114(45): 11962-11967.',\n",
       "  'id': 165,\n",
       "  'name': '92-Uda-CellsigDyn'},\n",
       " {'description': 'See details in Nojima et al. (2017) Scientific Reports, 7(1): 9269.',\n",
       "  'id': 166,\n",
       "  'name': '95-Nojima-CellMorphology'},\n",
       " {'description': 'See details in Kamimura et al. (2016) PNAS, 113(16): 4356-4361.',\n",
       "  'id': 167,\n",
       "  'name': '96-Kamimura-ChemotaxisDyn'},\n",
       " {'description': 'See details in Isshiki et al. (2014) Nat Commun, 5: 4742.',\n",
       "  'id': 201,\n",
       "  'name': '89-Isshiki-SpineDyn'}]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ssbd_list = get_all_projects(SSBD_OMERO_URL)\n",
    "ssbd_list"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## function get_all_proj_attrib() returns a list of attributes given a URL and a project list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_all_proj_attrib(base_url, project_list):\n",
    "    dataset_attributes_url = \"{base}/webclient/api/annotations/?type=map&project={proj}\"\n",
    "    proj_attrib = []\n",
    "    for proj in project_list:\n",
    "        qs = {'base': base_url, 'proj': proj['id']}\n",
    "        url = dataset_attributes_url.format(**qs)\n",
    "        print(url)\n",
    "        r = requests.session().get(url)\n",
    "        if r.status_code == 200:\n",
    "            metadata = r.json()\n",
    "        else:\n",
    "            quit()\n",
    "        attribute = proj\n",
    "        for a in metadata['annotations']:\n",
    "            for v in a['values']:\n",
    "                attribute[v[0]] = v[1]\n",
    "        proj_attrib.append(attribute)\n",
    "    return(proj_attrib)\n",
    "    \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get all the attributes of all the projects in SSBD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=68\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=11\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=1\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=101\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=4\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=67\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=61\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=71\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=64\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=62\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=52\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=51\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=69\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=10\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=53\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=66\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=63\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=5\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=54\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=65\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=2\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=59\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=9\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=106\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=105\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=104\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=70\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=12\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=3\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=55\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=103\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=56\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=57\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=58\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=8\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=6\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=60\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=102\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=7\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=152\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=154\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=158\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=159\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=156\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=171\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=168\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=169\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=170\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=151\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=173\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=153\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=155\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=157\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=160\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=161\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=162\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=172\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=163\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=164\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=165\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=166\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=167\n",
      "http://ssbd.qbic.riken.jp/image/webclient/api/annotations/?type=map&project=201\n"
     ]
    }
   ],
   "source": [
    "ssbd_attrib = get_all_proj_attrib(SSBD_OMERO_URL, ssbd_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'description': 'See details in Inoue and Kondo (2016) Scientific Reports, 6, 33689.',\n",
       "  'id': 68,\n",
       "  'name': '38-Inoue-ShellStruct',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Shigeru Kondo',\n",
       "  'Organization': 'Osaka University',\n",
       "  'Department': 'Graduate School of Frontier Bioscience',\n",
       "  'Laboratory': 'Laboratory of Pattern Formation',\n",
       "  'Contributers': 'Shinya Inoue, Shigeru Kondo',\n",
       "  'Organism': 'Damesites cf. damesi',\n",
       "  'PubMed': 'http://www.ncbi.nlm.nih.gov/pubmed/27640361'},\n",
       " {'description': 'See details in Takayama and Onami. (2016) Cell Reports, 15(3), 625-637',\n",
       "  'id': 11,\n",
       "  'name': '12-Takayama-Oocyte',\n",
       "  'License': 'CC BY-NC-SA',\n",
       "  'Contactname': 'Shuichi Onami',\n",
       "  'Organization': 'RIKEN',\n",
       "  'Department': 'Quantitative Biology Center',\n",
       "  'Laboratory': 'Laboratory for Developmental Dynamics',\n",
       "  'Contributers': 'Jun Takayama, Shuichi Onami',\n",
       "  'Organism': 'C. elegans',\n",
       "  'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/27068469'},\n",
       " {'description': 'See details in Kyoda et al. (2013) Nucleic Acids Res 41, D732-D737',\n",
       "  'id': 1,\n",
       "  'name': '2-Kyoda-WormEmbryoRNAi',\n",
       "  'License': 'CC BY-NC-SA',\n",
       "  'Contactname': 'Shuichi Onami',\n",
       "  'Organization': 'RIKEN',\n",
       "  'Department': 'Quantitative Biology Center',\n",
       "  'Laboratory': 'Laboratory for Developmental Dynamics',\n",
       "  'Contributers': 'Eru Adachi, Eriko Masuda, Yoko Nagai, Yoko Suzuki, Taeko Oguro, Koji Kyoda, Shuichi Onami',\n",
       "  'Organism': 'C. elegans',\n",
       "  'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/23172286'},\n",
       " {'description': 'See details in Azuma and Onami (2017) BMC Bioinformatics, 18(1): 307.',\n",
       "  'id': 101,\n",
       "  'name': '20-Azuma-WormMembrane',\n",
       "  'License': 'CC BY-NC-SA',\n",
       "  'Contactname': 'Shuichi Onami',\n",
       "  'Organization': 'RIKEN',\n",
       "  'Department': 'Quantitative Biology Center',\n",
       "  'Laboratory': 'Laboratory for Developmental Dynamics',\n",
       "  'Contributers': 'Yusuke Azuma, Shuichi Onami',\n",
       "  'Organism': 'Caenorhabditis elegans',\n",
       "  'PubMed': 'http://www.ncbi.nlm.nih.gov/pubmed/28629355'},\n",
       " {'description': 'See details in Toyoshima et al. (2016) PLoS Computational Biology, 12(6): e1004970',\n",
       "  'id': 4,\n",
       "  'name': '11-Toyoshima-NeuAct',\n",
       "  'License': 'CC BY-NC-SA',\n",
       "  'Contactname': 'Yu Toyoshima',\n",
       "  'Organization': 'The University of Tokyo',\n",
       "  'Department': 'Department of Biological Sciences',\n",
       "  'Laboratory': 'Iino Laboratory',\n",
       "  'Contributers': 'Yu Toyoshima, Terumasa Tokunaga, Osamu Hirose, Manami Kanamori, Takayuki Teramoto, Moon Sun Jang, Sayuri Kuge, Takeshi Ishihara, Ryo Yoshida, Yuichi Iino',\n",
       "  'Organism': 'C. elegans',\n",
       "  'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/27271939'},\n",
       " {'description': 'See details in Arata et al. (2016) Cell Reports, 16, 1-13.',\n",
       "  'id': 67,\n",
       "  'name': '37-Arata-MolDynPar2',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Yukinobu Arata, Yasushi Sako',\n",
       "  'Organization': 'RIKEN',\n",
       "  'Department': 'Wako',\n",
       "  'Laboratory': 'Cellular Informatics Laboratory',\n",
       "  'Contributers': 'Yukinobu Arata, Michio Hiroshima, Chan-Gi Pack, Ravikrishna Ramanujam, Fumio Motegi, Kenichi Nakazato, Yuki Shindo, Paul W. Wiseman, Hitoshi Sawa, Tetsuya J. Kobayashi, Hugo B. Brandão, Tatsuo Shibata, Yasushi Sako',\n",
       "  'Organism': 'Caenorhabditis elegans',\n",
       "  'PubMed': 'http://www.ncbi.nlm.nih.gov/pubmed/27524610'},\n",
       " {'description': 'See details in Matsuda et al. (2012) Sci. Signal., 5, ra31.',\n",
       "  'id': 61,\n",
       "  'name': '31-Matsuda-SigDyn',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Miki Ebisuya',\n",
       "  'Organization': 'RIKEN',\n",
       "  'Department': 'Quantitative Biology Center',\n",
       "  'Laboratory': 'Laboratory for Reconstitutive Developmental Biology',\n",
       "  'Contributers': 'Mitsuhiro Matsuda, Makito Koga, Eisuke Nishida, Miki Ebisuya',\n",
       "  'Organism': 'Cricetulus griseus, Canis lupus familiaris',\n",
       "  'PubMed': 'http://www.ncbi.nlm.nih.gov/pubmed/22510469'},\n",
       " {'description': 'See details in Takai et al. (2015) PNAS, 112(14): 4352-4356.',\n",
       "  'id': 71,\n",
       "  'name': '43-Takai-SubcellStructONL',\n",
       "  'License': 'CC BY-NC-SA',\n",
       "  'Contactname': 'Yasushi Okada',\n",
       "  'Organization': 'RIKEN',\n",
       "  'Department': 'Quantitative Biology Center',\n",
       "  'Laboratory': 'Laboratory for Cell Polarity Regulation',\n",
       "  'Contributers': 'Akira Takai, Masahiro Nakano, Kenta Saito, Remi Haruno, Tomonobu M. Watanabe, Tatsuya Ohyanagi, Takashi Jin, Yasushi Okada, Takeharu Nagai',\n",
       "  'Organism': 'Canis lupus familiaris',\n",
       "  'PubMed': 'http://www.ncbi.nlm.nih.gov/pubmed/25831507'},\n",
       " {'description': 'See details in Hayashi and Okada (2015) Mol Biol Cell, 26(9): 1743-1751.',\n",
       "  'id': 64,\n",
       "  'name': '34-Hayashi-CellDynSDSRM',\n",
       "  'License': 'CC BY-NC-SA',\n",
       "  'Contactname': 'Yasushi Okada',\n",
       "  'Organization': 'RIKEN',\n",
       "  'Department': 'Quantitative Biology Center',\n",
       "  'Laboratory': 'Laboratory for Cell Polarity Regulation',\n",
       "  'Contributers': 'Shinichi Hayashi, Yasushi Okada',\n",
       "  'Organism': 'Canis lupus familiaris',\n",
       "  'PubMed': 'http://www.ncbi.nlm.nih.gov/pubmed/25717185'},\n",
       " {'description': 'See details in Matsuda et al. (2015) Nat. Commun., 6, 6195',\n",
       "  'id': 62,\n",
       "  'name': '32-Matsuda-SigDyn',\n",
       "  'License': 'CC BY-NC-SA',\n",
       "  'Contactname': 'Miki Ebisuya',\n",
       "  'Organization': 'RIKEN',\n",
       "  'Department': 'Quantitative Biology Center',\n",
       "  'Laboratory': 'Laboratory for Reconstitutive Developmental Biology',\n",
       "  'Contributers': 'Mitsuhiro Matsuda, Makito Koga, Knut Woltjen, Eisuke Nishida, Miki Ebisuya',\n",
       "  'Organism': 'Cricetulus griseus',\n",
       "  'PubMed': 'http://www.ncbi.nlm.nih.gov/pubmed/25652697'},\n",
       " {'description': 'See details in Kondo and Hayashi (2013) Nature, 494, 125-129.',\n",
       "  'id': 52,\n",
       "  'name': '22-Kondo-CellDyn',\n",
       "  'License': 'CC BY-NC-SA',\n",
       "  'Contactname': 'Takefumi Kondo',\n",
       "  'Organization': 'RIKEN',\n",
       "  'Department': 'Center for Developmental Biology',\n",
       "  'Laboratory': 'Laboratory for Morphogenetic Signaling',\n",
       "  'Contributers': 'Takefumi Kondo, Shigeo Hayashi',\n",
       "  'Organism': 'Drosophila melanogaster',\n",
       "  'PubMed': 'http://www.ncbi.nlm.nih.gov/pubmed/23334416'},\n",
       " {'description': 'See details in Yasui et al. (2014) PLoS Computational Biology, 10(9): e1003817.',\n",
       "  'id': 51,\n",
       "  'name': '21-Yasui-MolDynPTEN',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Satomi Matsuoka',\n",
       "  'Organization': 'Osaka University',\n",
       "  'Department': 'Graduate School of Frontier Biosciences',\n",
       "  'Laboratory': 'Laboratories for Nanobiology',\n",
       "  'Contributers': 'Masato Yasui, Satomi Matsuoka, Masahiro Ueda',\n",
       "  'Organism': 'Dictyostelium discoideum',\n",
       "  'PubMed': 'http://www.ncbi.nlm.nih.gov/pubmed/25211206'},\n",
       " {'description': 'See details in Shibata et al. (2012) J. Cell Sci., 125: 5138-5150.',\n",
       "  'id': 69,\n",
       "  'name': '39-Shibata-MolDynPTEN',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Tatsuo Shibata',\n",
       "  'Organization': 'RIKEN',\n",
       "  'Department': 'Quantitative Biology Center',\n",
       "  'Laboratory': 'Laboratories for Physical Biology',\n",
       "  'Contributers': 'Tatsuo Shibata, Masatoshi Nishikawa, Satomi Matsuoka, Masahiro Ueda',\n",
       "  'Organism': 'Dictyostelium discoideum',\n",
       "  'PubMed': 'http://www.ncbi.nlm.nih.gov/pubmed/22899720'},\n",
       " {'description': 'See details in Komatsuzaki et al. (2015) Small, 11(12):1396-1401',\n",
       "  'id': 10,\n",
       "  'name': '10-Komatsuzaki-MolDyn',\n",
       "  'License': 'CC BY-NC-SA',\n",
       "  'Contactname': 'Takashi Jin',\n",
       "  'Organization': 'RIKEN',\n",
       "  'Department': 'Quantitative Biology Center',\n",
       "  'Laboratory': 'Laboratory for Nano-Bio Probes',\n",
       "  'Contributers': 'Akihiro Komatsuzaki, Tatsuya Ohyanagi, Yoshikazu Tsukasaki, Yukihiro Miyanaga, Masahiro Ueda, Takashi Jin',\n",
       "  'Organism': 'D. discoideum',\n",
       "  'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/25504902'},\n",
       " {'description': 'See details in Shirai et al. (2016) PNAS, 113(1): E81-E90.',\n",
       "  'id': 53,\n",
       "  'name': '23-Shirai-ESCellTissue',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Michiko Mandai',\n",
       "  'Organization': 'RIKEN',\n",
       "  'Department': 'Center for Developmental Biology',\n",
       "  'Laboratory': 'Laboratory for Retinal Regeneration',\n",
       "  'Contributers': 'Hiroshi Shirai, Michiko Mandaia, Keizo Matsushitaa, Atsushi Kuwaharac, Shigenobu Yonemuraf, Tokushige Nakanod, Juthaporn Assawachananonta, Toru Kimurac, Koichi Saitoe, Hiroko Terasakib, Mototsugu Eirakug, Yoshiki Sasaid, Masayo Takahashi',\n",
       "  'Organism': 'Homo sapiens',\n",
       "  'PubMed': 'http://www.ncbi.nlm.nih.gov/pubmed/26699487'},\n",
       " {'description': 'See details in Torisawa et al. (2016) Biophysical Journal, 111(2): 373-385.',\n",
       "  'id': 66,\n",
       "  'name': '36-Torisawa-MolDynEg5',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Kazuhiro Oiwa',\n",
       "  'Organization': 'National Institute of Information and Communications Technology',\n",
       "  'Department': 'Advanced ICT Research Institute',\n",
       "  'Laboratory': '-',\n",
       "  'Contributers': 'Takayuki Torisawa, Daisuke Taniguchi, Shuji Ishihara, Kazuhiro Oiwa',\n",
       "  'Organism': '-',\n",
       "  'PubMed': 'http://www.ncbi.nlm.nih.gov/pubmed/27463139'},\n",
       " {'description': 'See details in Yamao et al. (2015) Scientific Reports, 5: 17527.',\n",
       "  'id': 63,\n",
       "  'name': '33-Yamao-MolDynRho',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Naoki Honda',\n",
       "  'Organization': 'Kyoto University',\n",
       "  'Department': 'Graduate School of Informatics',\n",
       "  'Laboratory': '-',\n",
       "  'Contributers': 'Masataka Yamao, Honda Naoki, Katsuyuki Kunida, Kazuhiro Aoki, Michiyuki Matsuda, Shin Ishii',\n",
       "  'Organism': 'Homo sapiens',\n",
       "  'PubMed': 'http://www.ncbi.nlm.nih.gov/pubmed/26634649'},\n",
       " {'description': 'See details in Kunida et al. (2012) Journal of Cell Science, 125(10): 2381-2392',\n",
       "  'id': 5,\n",
       "  'name': '18-Kunida-MolDynRho',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Kazuhiro Aoki',\n",
       "  'Organization': 'Kyoto University',\n",
       "  'Department': 'Graduate School of Medicine',\n",
       "  'Laboratory': 'Imaging Platform for Spatio-Temporal Information',\n",
       "  'Contributers': 'Katsuyuki Kunida, Michiyuki Matsuda, Kazuhiro Aoki',\n",
       "  'Organism': 'H. sapiens',\n",
       "  'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/22344265'},\n",
       " {'description': 'See details in Kanemura et al. (2013) Scientific Reports, 3(2334): 2334.',\n",
       "  'id': 54,\n",
       "  'name': '24-Kanemura-iPSCellDyn',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Shin Kawamata',\n",
       "  'Organization': 'Foundation for Biomedical Research and Innovation',\n",
       "  'Department': 'Research & Development Center for Cell Therapy',\n",
       "  'Laboratory': 'Laboratory for Retinal Regeneration',\n",
       "  'Contributers': 'Hoshimi Kanemura, Masahiro J. Go, Naoki Nishishita, Noriko Sakai, Hiroyuki Kamao, Yoji Sato, Masayo Takahashi, Shin Kawamata',\n",
       "  'Organism': 'Homo sapiens',\n",
       "  'PubMed': 'http://www.ncbi.nlm.nih.gov/pubmed/23903667'},\n",
       " {'description': 'See details in Yonemura (2014) PLoS ONE, 9(11): e112922.',\n",
       "  'id': 65,\n",
       "  'name': '35-Yonemura-CellDyn',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Shigenobu Yonemura',\n",
       "  'Organization': 'RIKEN',\n",
       "  'Department': 'Center for Developmental Biology',\n",
       "  'Laboratory': 'Electron Microscope Laboratory',\n",
       "  'Contributers': 'Shigenobu Yonemura',\n",
       "  'Organism': 'Homo sapiens',\n",
       "  'PubMed': 'http://www.ncbi.nlm.nih.gov/pubmed/25393292'},\n",
       " {'description': 'See details in Watabe et al. (2015) PLoS One, 10(7): e0130089',\n",
       "  'id': 2,\n",
       "  'name': '13-Watabe-MolDyn',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Koichi Takahashi',\n",
       "  'Organization': 'RIKEN',\n",
       "  'Department': 'Quantitative Biology Center',\n",
       "  'Laboratory': 'Laboratory for Biochemical Simulation',\n",
       "  'Contributers': 'Masaki Watabe, Satya N. V. Arjunan, Seiya Fukushima, Kazunari Iwamoto, Jun Kozuka, Satomi Matsuoka, Yuki Shindo, Masahiro Ueda, Koichi Takahashi',\n",
       "  'Organism': 'D. discoideum',\n",
       "  'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/26147508'},\n",
       " {'description': 'See details in Ke et al. (2016) Cell Reports, 14: 2718-2732.',\n",
       "  'id': 59,\n",
       "  'name': '29-Ke-SeeDB2',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Takeshi Imai',\n",
       "  'Organization': 'RIKEN',\n",
       "  'Department': 'Center for Developmental Biology',\n",
       "  'Laboratory': 'Laboratory for Sensory Circuit Formation',\n",
       "  'Contributers': 'Meng-Tsen Ke, Yasuhiro Nakai, Satoshi Fujimoto, Rie Takayama, Shuhei Yoshida, Tomoya S. Kitajima, Makoto Sato, Takeshi Imai',\n",
       "  'Organism': 'Mus musculus',\n",
       "  'PubMed': 'http://www.ncbi.nlm.nih.gov/pubmed/26972009'},\n",
       " {'description': 'See details in Bashar et al. (2012) PLoS ONE 7, e35550.',\n",
       "  'id': 9,\n",
       "  'name': '8-Bashar-MouseEmbryo',\n",
       "  'License': 'CC BY-NC-SA',\n",
       "  'Contactname': 'Md. Khayrul Bashar',\n",
       "  'Organization': 'The University of Tokyo',\n",
       "  'Department': 'Institute of Industrial Science',\n",
       "  'Laboratory': 'Laboratory for Quantitative Biology',\n",
       "  'Contributers': 'Md. Khayrul Bashar, Koji Komatsu, Toshihiko Fujimori, Tetsuya J. Kobayashi',\n",
       "  'Organism': 'M. musculus',\n",
       "  'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/22590505'},\n",
       " {'description': 'See details in Susaki et al. (2014) Cell, 157(3): 726–739.',\n",
       "  'id': 106,\n",
       "  'name': '48-Susaki-MouseBrainCUBIC',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Hiroki R Ueda',\n",
       "  'Organization': 'RIKEN',\n",
       "  'Department': 'Quantitative Biology Center',\n",
       "  'Laboratory': 'Laboratory for Synthetic Biology',\n",
       "  'Contributers': 'Etsuo A. Susaki, Kazuki Tainaka, Dimitri Perrin, Fumiaki Kishino, Takehiro Tawara, Tomonobu M. Watanabe, Chihiro Yokoyama, Hirotaka Onoe, Megumi Eguchi, Shun Yamaguchi, Takaya Abe, Hiroshi Kiyonari, Yoshihiro Shimizu, Atsushi Miyawaki, Hideo Yokota, Hiroki R. Ueda',\n",
       "  'Organism': 'Mus musculus',\n",
       "  'PubMed': 'http://www.ncbi.nlm.nih.gov/pubmed/24746791'},\n",
       " {'description': 'See details in Susaki et al. (2015) Nature Protocols, 10: 1709–1727.',\n",
       "  'id': 105,\n",
       "  'name': '49-Susaki-MouseBrainCUBIC',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Hiroki R Ueda',\n",
       "  'Organization': 'RIKEN',\n",
       "  'Department': 'Quantitative Biology Center',\n",
       "  'Laboratory': 'Laboratory for Synthetic Biology',\n",
       "  'Contributers': 'Etsuo A Susaki, Kazuki Tainaka, Dimitri Perrin, Hiroko Yukinaga, Akihiro Kuno, Hiroki R Ueda',\n",
       "  'Organism': 'Mus musculus',\n",
       "  'PubMed': 'http://www.ncbi.nlm.nih.gov/pubmed/26448360'},\n",
       " {'description': 'See details in Tainaka et al. (2014) Cell, 159(4): 911–924.',\n",
       "  'id': 104,\n",
       "  'name': '50-Tainaka-MouseBodyCUBIC',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Hiroki R Ueda',\n",
       "  'Organization': 'RIKEN',\n",
       "  'Department': 'Quantitative Biology Center',\n",
       "  'Laboratory': 'Laboratory for Synthetic Biology',\n",
       "  'Contributers': 'Kazuki Tainaka, Shimpei I. Kubota, Takeru Q. Suyama, Etsuo A. Susaki, Dimitri Perrin, Maki Ukai-Tadenuma, Hideki Ukai, Hiroki R. Ueda',\n",
       "  'Organism': 'Mus musculus',\n",
       "  'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/25417165'},\n",
       " {'description': 'See details in Sakakibara et al. (2015) Nat Commun. 2013; 4:2212.',\n",
       "  'id': 70,\n",
       "  'name': '40-Sakakibara-KTDyn',\n",
       "  'License': 'CC BY-NC-SA',\n",
       "  'Contactname': 'Tomoya Kitajima',\n",
       "  'Organization': 'RIKEN',\n",
       "  'Department': 'Center for Developmental Biology',\n",
       "  'Laboratory': 'Laboratory for Chromosome Segregation',\n",
       "  'Contributers': 'Yogo Sakakibara, Tomoya Kitajima',\n",
       "  'Organism': 'Mus musculus',\n",
       "  'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/26130582'},\n",
       " {'description': 'See details in Ochiai et al. (2015) Nucleic Acids Research, 43(19): e127',\n",
       "  'id': 12,\n",
       "  'name': '14-Ochiai-MolDynMS2',\n",
       "  'License': 'CC BY-NC',\n",
       "  'Contactname': 'Hiroshi Ochiai',\n",
       "  'Organization': 'Hiroshima University',\n",
       "  'Department': 'Research Center for the Mathematics on Chromatin Live Dynamics (RcMcD)',\n",
       "  'Contributers': 'Hiroshi Ochiai, Takeshi Sugiwara, Takashi Yamamoto',\n",
       "  'Organism': 'M. musculus',\n",
       "  'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/26092696'},\n",
       " {'description': 'See details in Harima et al. (2013) Cell Reports, 3(1): 1-7',\n",
       "  'id': 3,\n",
       "  'name': '15-Harima-GeneExp',\n",
       "  'License': 'CC BY-NC-SA',\n",
       "  'Contactname': 'Ryoichiro Kageyama',\n",
       "  'Organization': 'Kyoto University',\n",
       "  'Department': 'Institute for Virus Research',\n",
       "  'Contributers': 'Yukiko Harima, Yoshiki Takashima, Yuriko Ueda, Toshiyuki Ohtsuka, and Ryoichiro Kageyama',\n",
       "  'Organism': 'M. musculus',\n",
       "  'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/23219549'},\n",
       " {'description': 'See details in Ke et al. (2013) Nature Neuroscience, 16: 1154-1161.',\n",
       "  'id': 55,\n",
       "  'name': '25-Ke-SeeDB',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Takeshi Imai',\n",
       "  'Organization': 'RIKEN',\n",
       "  'Department': 'Center for Developmental Biology',\n",
       "  'Laboratory': 'Laboratory for Sensory Circuit Formation',\n",
       "  'Contributers': 'Meng-Tsen Ke, Satoshi Fujimoto, Takeshi Imai',\n",
       "  'Organism': 'Mus musculus',\n",
       "  'PubMed': 'http://www.ncbi.nlm.nih.gov/pubmed/23792946'},\n",
       " {'description': 'See details in Ichimura et al. (2016) Scientific Reports, 6: 37562.',\n",
       "  'id': 103,\n",
       "  'name': '53-Ichimura-CellDyn',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Hideaki Fujita',\n",
       "  'Organization': 'RIKEN',\n",
       "  'Department': 'Quantitative Biology Center',\n",
       "  'Laboratory': 'Laboratory for Comprehensive Bioimaging',\n",
       "  'Contributers': 'Taro Ichimura, Liang-da Chiu, Katsumasa Fujita, Hiroaki Machiyama, Tomoyuki Yamaguchi, Tomonobu M. Watanabe, Hideaki Fujita',\n",
       "  'Organism': 'Mus musculus',\n",
       "  'PubMed': 'http://www.ncbi.nlm.nih.gov/pubmed/27876845'},\n",
       " {'description': 'See details in Kinoshita et al. (2016) Retinal Cell Biology, 57(1): 153-161.',\n",
       "  'id': 56,\n",
       "  'name': '26-Kinoshita-iPSCellDyn',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Kiyoshi Suzuma',\n",
       "  'Organization': 'Nagasaki University',\n",
       "  'Department': 'Department of Ophthalmology and Visual Sciences, Graduate School of Biomedical Sciences',\n",
       "  'Laboratory': '-',\n",
       "  'Contributers': 'Hirofumi Kinoshita, Kiyoshi Suzuma, Jun Kaneko, Michiko Mandai, Takashi Kitaoka, Masayo Takahashi',\n",
       "  'Organism': 'Mus musculus',\n",
       "  'PubMed': 'http://www.ncbi.nlm.nih.gov/pubmed/26780319'},\n",
       " {'description': 'See details in Iwasaki et al. (2016) PLoS ONE, 11(7): e0158282.',\n",
       "  'id': 57,\n",
       "  'name': '27-Iwasaki-iPSCellDyn',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Masayo Takahashi',\n",
       "  'Organization': 'RIKEN',\n",
       "  'Department': 'Center for Developmental Biology',\n",
       "  'Laboratory': 'Laboratory for Retinal Regeneration',\n",
       "  'Contributers': 'Yuko Iwasaki, Sunao Sugita, Michiko Mandai, Shigenobu Yonemura, Akishi Onishi, Shin-ichiro Ito, Manabu Mochizuki Kyoko Ohno-Matsui, Masayo Takahashi',\n",
       "  'Organism': 'Mus musculus',\n",
       "  'PubMed': 'http://www.ncbi.nlm.nih.gov/pubmed/27385038'},\n",
       " {'description': 'See details in Assawachananont et al. (2014) Stem Cell Reports, 2(5): 662-674.',\n",
       "  'id': 58,\n",
       "  'name': '28-Assawachananont-iPSCellTissue',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Michiko Mandai',\n",
       "  'Organization': 'RIKEN',\n",
       "  'Department': 'Center for Developmental Biology',\n",
       "  'Laboratory': 'Laboratory for Retinal Regeneration',\n",
       "  'Contributers': 'Juthaporn Assawachananont, Michiko Mandai, Satoshi Okamoto, Chikako Yamada, Mototsugu Eiraku, Shigenobu Yonemura, Yoshiki Sasai, Masayo Takahashi',\n",
       "  'Organism': 'Mus musculus',\n",
       "  'PubMed': 'http://www.ncbi.nlm.nih.gov/pubmed/24936453'},\n",
       " {'description': 'See details in Tanaka (2014) Micromachines, 5(2): 289-299',\n",
       "  'id': 8,\n",
       "  'name': '19-Tanaka-FluidDyn',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Yo Tanaka',\n",
       "  'Organization': 'RIKEN',\n",
       "  'Department': 'Quantitative Biology Center',\n",
       "  'Laboratory': 'Laboratory for Integrated Biodevice',\n",
       "  'Contributers': 'Yo Tanaka',\n",
       "  'Organism': 'NA'},\n",
       " {'description': 'See details in Aoki et al. (2013) Molecular Cell, 52(4): 529-540',\n",
       "  'id': 6,\n",
       "  'name': '17-Aoki-MolDyn',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Kazuhiro Aoki',\n",
       "  'Organization': 'Kyoto University',\n",
       "  'Department': 'Graduate School of Medicine',\n",
       "  'Laboratory': 'Imaging Platform for Spatio-Temporal Information',\n",
       "  'Contributers': 'Kazuhiro Aoki, Yuka Kumagai, Atsuro Sakurai, Naoki Komatsu, Yoshihisa Fujita, Clara Shionyu, Michiyuki Matsuda',\n",
       "  'Organism': 'R. norvegicus',\n",
       "  'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/24140422'},\n",
       " {'description': 'See details in Inomata et al. (2013) Cell, 153(6): 1296-1311.',\n",
       "  'id': 60,\n",
       "  'name': '30-Inomata-CellDyn',\n",
       "  'License': 'CC BY-NC-SA',\n",
       "  'Contactname': 'Hidehiko Inomata',\n",
       "  'Organization': 'RIKEN',\n",
       "  'Department': 'Center for Developmental Biology',\n",
       "  'Laboratory': 'Laboratory for Physical Biology',\n",
       "  'Contributers': 'Hidehiko Inomata, Tatsuo Shibata, Tomoko Haraguchi, Yoshiki Sasai',\n",
       "  'Organism': 'Xenopus laevis',\n",
       "  'PubMed': 'http://www.ncbi.nlm.nih.gov/pubmed/23746842'},\n",
       " {'description': 'See details in Suzuki et al. (2015) Genes to Cells, 21(4): 358-369.',\n",
       "  'id': 102,\n",
       "  'name': '52-Suzuki-FrogTailDyn',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Ken-ichi T. Suzuki',\n",
       "  'Organization': 'Hiroshima University',\n",
       "  'Department': 'Department of Mathematical and Life Sciences',\n",
       "  'Laboratory': 'Molecular Genetics Laboratory',\n",
       "  'Contributers': 'Miyuki Suzuki, Chiyo Takagi, Shinichirou Miura, Yuto Sakane, Makoto Suzuki, Tetsushi Sakuma, Naoaki Sakamoto, Tetsuya Endo, Yasuhiro Kamei, Yuko Sato, Hiroshi Kimura, Takashi Yamamoto, Naoto Ueno, Ken-ichi T. Suzuki',\n",
       "  'Organism': 'Xenopus laevis',\n",
       "  'PubMed': 'http://www.ncbi.nlm.nih.gov/pubmed/26914410'},\n",
       " {'description': 'See details in Tanaka and Fujita (2015) Sensors and Actuators B: Chemical, 210(1): 267-272',\n",
       "  'id': 7,\n",
       "  'name': '16-Tanaka-FluidDyn',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Yo Tanaka',\n",
       "  'Organization': 'RIKEN',\n",
       "  'Department': 'Quantitative Biology Center',\n",
       "  'Laboratory': 'Laboratory for Integrated Biodevice',\n",
       "  'Contributers': 'Yo Tanaka, Hideaki Fujita',\n",
       "  'Organism': 'M. musculus'},\n",
       " {'description': 'See details in Herawati et al. (2016) J. Cell Biol., 214(5): 571-586.',\n",
       "  'id': 152,\n",
       "  'name': '57-Herawati-CellDyn',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Sachiko Tsukita, Shuji Ishihara',\n",
       "  'Organization': 'Osaka University',\n",
       "  'Department': 'Graduate School of Frontier Biosciences and Medicine',\n",
       "  'Laboratory': 'Laboratory of Biological Science',\n",
       "  'Contributers': 'Elisa Herawati, Daisuke Taniguchi, Hatsuho Kanoh, Kazuhiro Tateishi, Shuji Ishihara, Sachiko Tsukita',\n",
       "  'Organism': 'M. musculus',\n",
       "  'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/27573463'},\n",
       " {'description': 'See details in Minegishi et al. (2017) Dev Cell, 40(5): 439-452.',\n",
       "  'id': 154,\n",
       "  'name': '75-Minegishi-NodeCell',\n",
       "  'License': 'CC BY-NC-SA',\n",
       "  'Contactname': 'Hiroshi Hamada',\n",
       "  'Organization': 'Osaka University',\n",
       "  'Department': 'Graduate School of Frontier Biosciences',\n",
       "  'Laboratory': 'Developmental Genetics Group',\n",
       "  'Contributers': 'Katsura Minegishi, Masakazu Hashimoto, Rieko Ajima, Katsuyoshi Takaoka, Kyosuke Shinohara, Yayoi Ikawa, Hiromi Nishimura, Andrew P. McMahon, Karl Willert, Yasushi Okada, Hiroshi Sasaki, Dongbo Shi, Toshihiko Fujimori, Toshihisa Ohtsuka, Yasunobu Igarashi, Terry P. Yamaguchi, Akihiko Shimono, Hidetaka Shiratori, Hiroshi Hamada',\n",
       "  'Organism': 'M. musculus',\n",
       "  'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/28292423'},\n",
       " {'description': 'See details in Yasugi et al. (2017) Scientific Reports, 7(1): 1619.',\n",
       "  'id': 158,\n",
       "  'name': '81-Yasugi-DmeCellDyn',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Takashi Nishimura',\n",
       "  'Organization': 'RIKEN',\n",
       "  'Department': 'Center for Biosystems Dynamics Research',\n",
       "  'Laboratory': 'Laboratory for Growth Control Signaling',\n",
       "  'Contributers': 'Tetsuo Yasugi, Takayuki Yamada, Takashi Nishimura',\n",
       "  'Organism': 'D. melanogaster',\n",
       "  'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/28487555'},\n",
       " {'description': 'See details in Ikeda et al. (2018) Nat Commun, 9(1): 1387.',\n",
       "  'id': 159,\n",
       "  'name': '82-Ikeda-GeneExp',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Takashi Ikeda, Keisuke Okita, Shinji Masui',\n",
       "  'Organization': 'Kyoto University',\n",
       "  'Department': 'Department of Life Science Frontiers, Center for iPS Cell Research and Application (CiRA)',\n",
       "  'Laboratory': 'Okita Lab',\n",
       "  'Contributers': 'Takashi Ikeda, Takafusa Hikichi, Hisashi Miura, Hirofumi Shibata, Kanae Mitsunaga, Yosuke Yamada, Knut Woltjen, Kei Miyamoto, Ichiro Hiratani, Yasuhiro Yamada, Akitsu Hotta, Takuya Yamamoto, Keisuke Okita, Shinji Masui',\n",
       "  'Organism': 'M. musculus',\n",
       "  'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/29643333'},\n",
       " {'description': 'See details in Takagi et al. (2017) Clinical Ophthalmology, 11: 1219-1225.',\n",
       "  'id': 156,\n",
       "  'name': '78-Takagi-CellDyn',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Seiji Takagi',\n",
       "  'Organization': 'Kobe City Medical Center general hospital',\n",
       "  'Department': 'Department of Ophthalmology',\n",
       "  'Laboratory': '-',\n",
       "  'Contributers': 'Seiji Takagi, Michiko Mandai, Noriko Miyamoto, Akihiro Nishida, Yasuhiko Hirami, Hirofumi Uyama, Midori Yamamoto, Masayo Takahashi, Goji Tomita, Yasuo Kurimoto',\n",
       "  'Organism': 'H. sapiens',\n",
       "  'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/28721006'},\n",
       " {'description': 'See details in Noguchi et al. (2015) Cell Reports, 13(12): 2679–2686.',\n",
       "  'id': 171,\n",
       "  'name': '47-Noguchi-CellDyn',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Masafumi Noguchi, Mitsuru Morimoto',\n",
       "  'Organization': 'RIKEN',\n",
       "  'Department': 'Center for Developmental Biology',\n",
       "  'Laboratory': 'Laboratory for Lung Development',\n",
       "  'Contributers': 'Masafumi Noguchi, Kenta Sumiyama, Mitsuru Morimoto',\n",
       "  'Organism': 'M. musculus',\n",
       "  'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/26711336'},\n",
       " {'description': 'See details in Jin et al. (2016) Med. Chem. Commun., 7(4): 623-631.',\n",
       "  'id': 168,\n",
       "  'name': '106-Jin-BreastTumorFlu',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Takashi Jin',\n",
       "  'Organization': 'RIKEN',\n",
       "  'Department': 'Quantitative Biology Center',\n",
       "  'Laboratory': 'Laboratory for Nano-Bio Probes',\n",
       "  'Contributers': 'Takashi Jin, Setsuko Tsuboi, Akihito Komatsuzaki, Yukio Imamura, Yoshinori Muranaka, Takao Sakata, Hidehiro Yasuda',\n",
       "  'Organism': 'M. musculus',\n",
       "  'PubMed': '-'},\n",
       " {'description': 'See details in Tsuboi et al. (2017) Chem. Commun., 53(68): 9450-9453.',\n",
       "  'id': 169,\n",
       "  'name': '107-Tsuboi-BreastTumorFlu',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Takashi Jin',\n",
       "  'Organization': 'RIKEN',\n",
       "  'Department': 'Center for Biosystems Dynamics Research',\n",
       "  'Laboratory': 'Laboratory for Nano-Bio Probes',\n",
       "  'Contributers': 'Setsuko Tsuboi, Akira Sasaki, Takao Sakata, Hidehiro Yasuda, Takashi Jin',\n",
       "  'Organism': 'M. musculus',\n",
       "  'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/28795717'},\n",
       " {'description': 'See details in Makino et al. (2016) FASEB Journal, 31(4): 1301-1322.',\n",
       "  'id': 170,\n",
       "  'name': '110-Makino-MembraneDyn',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Toshihide Kobayashi',\n",
       "  'Organization': 'University of Strasbourg',\n",
       "  'Department': 'Faculty of pharmacy',\n",
       "  'Laboratory': 'UMR 7021 CNRS',\n",
       "  'Contributers': 'Asami Makino, Mitsuhiro Abe, Reiko Ishitsuka, Motohide Murate, Takuma Kishimoto, Shota Sakai, Françoise Hullin-Matsuda, Yukiko Shimada, Takehiko Inaba, Hideyuki Miyatake, Hideko Tanaka, Atsushi Kurahashi, Chan-Gi Pack, Rinshi S. Kasai, Shuku Kubo, Nicole L. Schieber, Naoshi Dohmae, Naoya Tochio, Kyoji Hagiwara, Yutaka Sasaki, Yoko Aida, Fumihiro Fujimori, Takanori Kigawa, Kozo Nishibori, Robert G. Parton, Akihiro Kusumi, Yasushi Sako, Gregor Anderluh, Makoto Yamashita, Tetsuyuki Kobayashi, Peter Greimel, Toshihide Kobayashi',\n",
       "  'Organism': 'H. sapiens',\n",
       "  'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/27492925'},\n",
       " {'description': 'See details in Maekawa et al. (2015) Current Eye Research, 41(4): 558-568.',\n",
       "  'id': 151,\n",
       "  'name': '55-Maekawa-ESCellDyn',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Akishi Onishi',\n",
       "  'Organization': 'RIKEN',\n",
       "  'Department': 'Center for Developmental Biology',\n",
       "  'Laboratory': 'Laboratory for Retinal Regeneration',\n",
       "  'Contributers': 'Yuki Maekawa, Akishi Onishi, Keizo Matsushita, Naoshi Koide, Michiko Mandai, Kiyoshi Suzuma, Takashi Kitaoka, Atsushi Kuwahara, Chikafumi Ozone, Tokushige Nakano, Mototsugu Eiraku, Masayo Takahashi',\n",
       "  'Organism': 'M. musculus',\n",
       "  'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/25880804'},\n",
       " {'description': 'See details in Yano et al. (2013) J. Cell Biol., 203(4): 605-614.',\n",
       "  'id': 173,\n",
       "  'name': '59-Yano-MicrotubuleImmuno',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Sachiko Tsukita',\n",
       "  'Organization': 'Osaka University',\n",
       "  'Department': 'Graduate School of Frontier Biosciences and Medicine',\n",
       "  'Laboratory': 'Laboratory of Biological Science',\n",
       "  'Contributers': 'Tomoki Yano, Takeshi Matsui, Atsushi Tamura, Masami Uji, Sachiko Tsukita',\n",
       "  'Organism': 'M. musculus',\n",
       "  'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/24385485'},\n",
       " {'description': 'See details in Bansod et al. (2017) Development, 144(17): 3156-3167.',\n",
       "  'id': 153,\n",
       "  'name': '74-Bansod-NSCellHes5',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Toshiyuki Ohtsuka',\n",
       "  'Organization': 'Kyoto University',\n",
       "  'Department': 'Institute for Frontier Life and Medical Sciences',\n",
       "  'Laboratory': '-',\n",
       "  'Contributers': 'Shama Bansod, Ryoichiro Kageyama, Toshiyuki Ohtsuka',\n",
       "  'Organism': 'M. musculus',\n",
       "  'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/28851724'},\n",
       " {'description': 'See details in Aoki et al. (2017) Developmental Cell, 43(3): 305-317.',\n",
       "  'id': 155,\n",
       "  'name': '77-Aoki-CellDynERK',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Kazuhiro Aoki',\n",
       "  'Organization': 'National Institutes of Natural Sciences',\n",
       "  'Department': 'National Institute for Basic Biology',\n",
       "  'Laboratory': 'Division of Quantitative Biology',\n",
       "  'Contributers': 'Kazuhiro Aoki, Yohei Kondo, Honda Naoki, Toru Hiratsuka, Reina E. Itoh, Michiyuki Matsuda',\n",
       "  'Organism': 'M. musculus',\n",
       "  'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/29112851'},\n",
       " {'description': 'See details in Inomata et al. (2017) Chem Commun, 53(81): 11245-11248.',\n",
       "  'id': 157,\n",
       "  'name': '79-Inomata-HeLaCell',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Kohsuke Inomata, Takanori Kigawa',\n",
       "  'Organization': 'RIKEN',\n",
       "  'Department': 'Center for Biosystems Dynamics Research',\n",
       "  'Laboratory': 'Laboratory for Cellular Structural Biology ',\n",
       "  'Contributers': 'Kohsuke Inomata, Hajime Kamoshida, Masaomi Ikari, Yutaka Ito, Takanori Kigawa',\n",
       "  'Organism': 'H. sapiens',\n",
       "  'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/28960222'},\n",
       " {'description': 'See details in Fumoto et al. (2017) Development, 144(1): 151-162.',\n",
       "  'id': 160,\n",
       "  'name': '83-Fumoto-LungDevWnt',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Akira Kikuchi',\n",
       "  'Organization': 'Departments of Molecular Biology and Biochemistry, Graduate School of Medicine, Osaka University, 2-2 Yamadaoka, Suita 565-0871, Japan.',\n",
       "  'Department': 'Graduate School of Medicine',\n",
       "  'Laboratory': 'Departments of Molecular Biology and Biochemistry',\n",
       "  'Contributers': 'Katsumi Fumoto, Hisako Takigawa-Imamura, Kenta Sumiyama, Tomoyuki Kaneiwa, Akira Kikuchi',\n",
       "  'Organism': 'M. musculus',\n",
       "  'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/27913639'},\n",
       " {'description': 'See details in Ito et al. (2017) Nat Commun, 8(1): 1834-1849.',\n",
       "  'id': 161,\n",
       "  'name': '84-Ito-CellDyn',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Masatoshi Takeichi',\n",
       "  'Organization': 'RIKEN',\n",
       "  'Department': 'Center for Biosystems Dynamics Research',\n",
       "  'Laboratory': 'Laboratory for Cell Adhesion and Tissue Patterning',\n",
       "  'Contributers': 'Shoko Ito, Satoru Okuda, Masako Abe, Mari Fujimoto, Tetsuo Onuki, Tamako Nishimura, Masatoshi Takeichi',\n",
       "  'Organism': 'H. sapiens',\n",
       "  'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/29184140'},\n",
       " {'description': 'See details in Kosodo et al. (2017) Biochem Biophys Res Commun, 483(1): 94-100.',\n",
       "  'id': 162,\n",
       "  'name': '85-Kosodo-NeuralballDyn',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Yoichi Kosodo',\n",
       "  'Organization': 'Korea Brain Research Institute',\n",
       "  'Department': '-',\n",
       "  'Laboratory': 'Neural Regeneration Laboratory',\n",
       "  'Contributers': 'Yoichi Kosodo, Taeko Suetsugu, Tetsuya J. Kobayashi, Fumio Matsuzaki',\n",
       "  'Organism': 'M. musculus',\n",
       "  'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/28063930'},\n",
       " {'description': 'See details in Koeberle et al. (2017) Scientific Reports, 7(1): 13409.',\n",
       "  'id': 172,\n",
       "  'name': '87-CorneliaKoeverle-SpineDyn',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Shigeo Okabe',\n",
       "  'Organization': 'University of Tokyo',\n",
       "  'Department': 'Department of Cellular Neurobiology, Graduate School of Medicine',\n",
       "  'Laboratory': '-',\n",
       "  'Contributers': 'Solveigh Cornelia Koeberle, Shinji Tanaka, Toshihiko Kuriu, Hirohide Iwasaki, Andreas Koeberle, Alexander Schulz, Dario-Lucas Helbing, Yoko Yamagata, Helen Morrison, Shigeo Okabe',\n",
       "  'Organism': 'M. musculus',\n",
       "  'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/29042611'},\n",
       " {'description': 'See details in Ishiwata et al. (2017) Molecular Crystals and Liquid Crystals, 647(1): 127-150.',\n",
       "  'id': 163,\n",
       "  'name': '90-Ishiwata-SpindleDyn',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': \"Shin'ichi Ishiwata, Takeshi Itabashi\",\n",
       "  'Organization': 'Waseda University',\n",
       "  'Department': 'Department of Physics, Faculty of Science and Engineering',\n",
       "  'Laboratory': 'Ishiwata Lab.',\n",
       "  'Contributers': \"Shin'ichi Ishiwata, Makito Miyazaki, Katsuhiko Sato, Koutaro Nakagome,\\nSeine A. Shintani, Fuyu Kobirumaki-Shimozawa, Norio Fukuda, Kazuya Suzuki, Jun Takagi, Yuta Shimamotoe, Takeshi Itabashi\",\n",
       "  'Organism': 'X. laevis',\n",
       "  'PubMed': '-'},\n",
       " {'description': 'See details in Konagaya et al. (2017) Cell Reports, 21(9): 2628-2638.',\n",
       "  'id': 164,\n",
       "  'name': '91-Konagaya-AMPKactivity',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Kenta Terai',\n",
       "  'Organization': 'Kyoto University',\n",
       "  'Department': 'Graduate School of Biostudies',\n",
       "  'Laboratory': 'Laboratory of Bioimaging and Cell Signaling',\n",
       "  'Contributers': 'Yumi Konagaya, Kenta Terai, Yusuke Hirao, Kanako Takakura, Masamichi Imajo, Yuji Kamioka, Norio Sasaoka, Akira Kakizuka, Kenta Sumiyama, Tomoichiro Asano, Michiyuki Matsuda',\n",
       "  'Organism': 'H. sapiens',\n",
       "  'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/29186696'},\n",
       " {'description': 'See details in Uda et al. (2017) PNAS, 114(45): 11962-11967.',\n",
       "  'id': 165,\n",
       "  'name': '92-Uda-CellsigDyn',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Kazuhiro Aoki',\n",
       "  'Organization': 'National Institutes of Natural Sciences',\n",
       "  'Department': 'National Institute for Basic Biology',\n",
       "  'Laboratory': 'Division of Quantitative Biology',\n",
       "  'Contributers': 'Youichi Uda, Yuhei Goto, Shigekazu Oda, Takayuki Kohchi, Michiyuki Matsuda, Kazuhiro Aoki',\n",
       "  'Organism': 'H. sapiens',\n",
       "  'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/29078307'},\n",
       " {'description': 'See details in Nojima et al. (2017) Scientific Reports, 7(1): 9269.',\n",
       "  'id': 166,\n",
       "  'name': '95-Nojima-CellMorphology',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Hiroki R. Ueda',\n",
       "  'Organization': 'RIKEN',\n",
       "  'Department': 'Center for Biosystems Dynamics Research',\n",
       "  'Laboratory': 'Laboratory for Synthetic Biology',\n",
       "  'Contributers': 'Satoshi Nojima, Etsuo A. Susaki, Kyotaro Yoshida, Hiroyoshi Takemoto, Naoto Tsujimura, Shohei Iijima, Ko Takachi, Yujiro Nakahara, Shinichiro Tahara, Kenji Ohshima, Masako Kurashige, Yumiko Hori, Naoki Wada, Jun-ichiro Ikeda, Atsushi Kumanogoh, Eiichi Morii, Hiroki R. Ueda',\n",
       "  'Organism': 'H. sapiens',\n",
       "  'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/28839164'},\n",
       " {'description': 'See details in Kamimura et al. (2016) PNAS, 113(16): 4356-4361.',\n",
       "  'id': 167,\n",
       "  'name': '96-Kamimura-ChemotaxisDyn',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Masahiro Ueda',\n",
       "  'Organization': 'RIKEN',\n",
       "  'Department': 'Center for Biosystems Dynamics Research',\n",
       "  'Laboratory': 'Laboratory for Cell Signaling Dynamics',\n",
       "  'Contributers': 'Yoichiro Kamimuraa, Yukihiro Miyanagaa, Masahiro Ueda',\n",
       "  'Organism': 'D. discoideum AX2',\n",
       "  'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/27044073'},\n",
       " {'description': 'See details in Isshiki et al. (2014) Nat Commun, 5: 4742.',\n",
       "  'id': 201,\n",
       "  'name': '89-Isshiki-SpineDyn',\n",
       "  'License': 'CC BY',\n",
       "  'Contactname': 'Shigeo Okabe',\n",
       "  'Organization': 'University of Tokyo',\n",
       "  'Department': 'Department of Cellular Neurobiology, Graduate School of Medicine',\n",
       "  'Laboratory': '-',\n",
       "  'Contributers': 'Masaaki Isshiki, Shinji Tanaka, Toshihiko Kuriu, Katsuhiko Tabuchi, Toru Takumi, Shigeo Okabe',\n",
       "  'Organism': 'M. musculus',\n",
       "  'PubMed': 'https://www.ncbi.nlm.nih.gov/pubmed/25144834'}]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ssbd_attrib"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## function search_proj() returns a list of project with id, key, value given a list if attributes and a search term"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "def search_proj(proj_attrib, term):\n",
    "    result = []\n",
    "    for proj in proj_attrib:\n",
    "        for key, value in proj.items():\n",
    "            if key != 'id':\n",
    "                if term in value.lower():\n",
    "#                    print(\"project_id:\", proj['id'], key, \":\", value)\n",
    "                    result.append({'id': proj['id'], key : value})\n",
    "    return(result)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## searching SSBD-OMERO for the phrase \"elegans\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "ssbd_result = search_proj(ssbd_attrib, 'elegans')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'id': 11, 'Organism': 'C. elegans'},\n",
       " {'id': 1, 'Organism': 'C. elegans'},\n",
       " {'id': 101, 'Organism': 'Caenorhabditis elegans'},\n",
       " {'id': 4, 'Organism': 'C. elegans'},\n",
       " {'id': 67, 'Organism': 'Caenorhabditis elegans'}]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ssbd_result"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## searching SSBD-OMERO for the phrase \"koji\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'id': 1,\n",
       "  'Contributers': 'Eru Adachi, Eriko Masuda, Yoko Nagai, Yoko Suzuki, Taeko Oguro, Koji Kyoda, Shuichi Onami'},\n",
       " {'id': 9,\n",
       "  'Contributers': 'Md. Khayrul Bashar, Koji Komatsu, Toshihiko Fujimori, Tetsuya J. Kobayashi'}]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_proj(ssbd_attrib, 'koji')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Searching IDR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "IDR_URL = IDR_URL = \"https://idr.openmicroscopy.org\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get all the projects from IDR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'description': 'Publication Title\\nHuman lymphoid organ cDC2 and macrophages play complementary roles in the induction of T follicular helper responses\\n\\nExperiment Description\\nImaging mass cytometry of tonsil sections',\n",
       "  'id': 701,\n",
       "  'name': 'idr0054-segura-tonsilhyperion/experimentA'},\n",
       " {'description': 'Publication Title\\nUnbiased pattern analysis reveals highly diverse responses of cytoskeletal systems to cyclic straining\\n\\nExperiment Description\\nCharacterization of cytoskeletal systems (actin, microtubules, vimentin networks) to cyclic straining using confocal laser scanning microscopy, including experiments with nocodazole to depolymerize microtubules.',\n",
       "  'id': 505,\n",
       "  'name': 'idr0050-springer-cytoskeletalsystems/experimentA'},\n",
       " {'description': 'Publication Title\\nA microscopy data set of discrete spatial and temporal single molecule RNA expression in single cells\\n\\nExperiment Description\\nWe performed single molecule in-situ hybridization on yeast cells that have been exposed to osmotic stress and measured mRNA expression for two genes. The goal of this study was to generate data sets for single cell transcription modeling.',\n",
       "  'id': 503,\n",
       "  'name': 'idr0047-neuert-yeastmrna/experimentA'},\n",
       " {'description': 'Publication Title\\nVirtual nanoscopy: Generation of ultra-large high resolution electron microscopy maps\\n\\nExperiment Description\\n\\tUltra-large high resolution electron microscopy maps of the mouse glomerulus, mouse embryonic fibroblasts, human dendritic cells and a zebrafish embryo slice were generated using virtual nanoscopy.',\n",
       "  'id': 504,\n",
       "  'name': 'idr0053-faas-virtualnanoscopy/experimentA'},\n",
       " {'description': 'Publication Title\\nIn Toto Imaging and Reconstruction of Post-Implantation Mouse Development at the Single-Cell Level\\n\\nExperiment Description\\nLight-sheet imaging of post-implantation mouse development from E6.5 to E8.5',\n",
       "  'id': 502,\n",
       "  'name': 'idr0044-mcdole-tardislightsheet/experimentA'},\n",
       " {'description': 'Publication Title\\nNeuromesodermal Progenitors are a Conserved Source of Spinal Cord with Divergent Growth Dynamics\\n\\nExperiment Description\\nLightsheet imaging of tailbud from 18 somite stage with one frame per 2 minutes',\n",
       "  'id': 552,\n",
       "  'name': 'idr0051-fulton-tailbudlightsheet/experimentA'},\n",
       " {'description': 'Publication Title\\nTissue-based map of the human proteome.\\n\\nExperiment Description\\nMapping of protein expression across all major tissues and organs in the human body',\n",
       "  'id': 501,\n",
       "  'name': 'idr0043-uhlen-humanproteinatlas/experimentA'},\n",
       " {'description': 'Publication Title\\nDual spindle formation in zygotes keeps parental genomes apart in early mammalian embryos\\n\\nExperiment Description\\nCharacterization of mitotic spindle formation in live mouse zygotes by light-sheet microscopy.',\n",
       "  'id': 405,\n",
       "  'name': 'idr0045-reichmann-zygotespindle/experimentA'},\n",
       " {'description': 'Publication Title\\nEx vivo live cell tracking in kidney organoids using light sheet fluorescence microscopy\\n\\nExperiment Description\\nWe have adapted the mouse kidney rudiment assay to generate renal organoids. 24 hours after pelleting 100,000 mouse kidney cells, organoids had formed in the PDMS dishes. For the PNA vital stain test we used kidney cells from wildtype mice and for the tracking time series we used cells from the Wt1tm1Nhsn strain, expressing GFP together with the transcription factor Wt1. One organoid was embedded in a hydrogel matrix (1.5% Agarose + 3% gelatine). In the microscope chamber, the hydrogel cylinder was extruded from the capillary so that the hydrogel column containing the organoid was located outside the capillary and immersed in medium (phenol red-free MEM + 10% FBS, 1% pen/strep, 3ug/ml PNA-rh) at 37oC and with 5% CO2. We recorded z stacks of the spheroids from multiple angles every 30 mins to record the location of the GFP expressing cells in three dimensions over time. There are three time series. These are available in the subfolders \"Time PNA-rh vital\", \"Wt1-GFP PNA Time Series 1\" and \"Wt1-GFP PNA Time Series 2\".  The time series data was processed in Fiji by opening the green channel only, binning in X and Y by a factor of two and converting it to 8 bit for processing in TrackMate (v.2.8.2) where we used the LoG (Laplacian of Gaussian) Detector and LAP (Linear Assignment Problem) Tracker options. Tracking data generated from the time series using the Track Mate plug in for Fiji is also provided in the time series data sets as \"Nspots vs Time.txt\", \"Track statistics.txt\", \"Links in tracks statistics.txt\" and \"Spots in tracks statistics.txt\". The data was analysed and plotted via the Matlab script provided as \"TrackMateDataAnalysis.txt\" in conjunction with the functions \"NSpotsFunction.txt\", \"SpotsInTracksFunction.txt\", \"TrackStatisticsFunction.txt\", \"Violin.txt\" and \"violinplot.txt\".',\n",
       "  'id': 353,\n",
       "  'name': 'idr0038-held-kidneylightsheet/experimentC'},\n",
       " {'description': 'Publication Title\\nHigh resolution imaging reveals heterogeneity in chromatin states between cells that is not inherited through cell division.\\n\\nExperiment Description\\nTo monitor the organisation of chromatin in live cells were engineered with fluorescent fusion proteins which recognize specific operator sequences to tag pairs of syntenic gene loci. The separation of these loci was then tracked in three dimensions over time using fluorescence microscopy. Data generated for this manuscript are made accessible here. This includes the movies generated from yeast strains with different genomic separations. These are available within subfolders titled \"Colocalising\", \"Genomic separation 100kb\", \"Genomic separation 25kb\" etc. Tracking data generated from these movies is also available as idr0027-AnalysisAllData.csv in the Attachments section below. This includes x, y and z co-ordinates of foci in the red and green channels following two stages of correction for channel alignment, the distance between the red and green foci in nanometers,  the offset of the red channel foci with respect to the green channel foci in x, y, and z. The maximum intensity, background intensity and standard deviation of the background intensity for both the red and green channels. The contrast (maximum spot intensity - nuclear background intensity) for each channel. The name originally assigned to the video, the genomic separation of the strain in kb. The string in column AE starts with the letter assigned to the video in supplementary figure 2. Some videos include two cells. In these cases the higher cell (on the y axis) is indicated as \"upper\" and the other as \"lower\". One of the two cells can often be assigned as the mother and the other as daughter. This is indicated as m or d in column AF. The systematic name of the strain is indicated in column AG.',\n",
       "  'id': 151,\n",
       "  'name': 'idr0027-dickerson-chromatin/experimentA'},\n",
       " {'description': 'Publication Title\\nTiming of gene expression in a cell-fate decision system\\n\\nExperiment Description\\nTime-lapse with addition of 1uM alpha-factor at time 0',\n",
       "  'id': 401,\n",
       "  'name': 'idr0040-aymoz-singlecell/experimentA'},\n",
       " {'description': 'Publication Title\\nExperimental and computational framework for a dynamic protein atlas of human cell division\\n\\nExperiment Description\\nImage data used to build the mitotic cell atlas. This consists of three subsets: raw images, segmentation masks for the landmarks and protein concentration maps.',\n",
       "  'id': 404,\n",
       "  'name': 'idr0041-cai-mitoticatlas/experimentA'},\n",
       " {'description': 'Publication Title\\nA Deep-learning classifier identifies patients with clinical heart failure using whole-slide images of H&E tissue\\n\\nExperiment Description\\nHistopathology sub-image from patient whole-slide images from patients with end-stage clinical heart failure or cadaveric donor hearts from patients without heart failure.',\n",
       "  'id': 402,\n",
       "  'name': 'idr0042-nirschl-wsideeplearning/experimentA'},\n",
       " {'description': 'Publication Title\\nEx vivo live cell tracking in kidney organoids using light sheet fluorescence microscopy\\n\\nStudy Description\\nWe have adapted the mouse kidney rudiment assay to generate renal organoids. We have immunostained endpoint fixed organoids and E13.5 embryonic kidneys and imaged them with a light sheet fluorescent microscope. The data generated for this manuscript is made accessible here and includes immunofluorescence staining after end point fixation revealing that organotypic structures develop within the organoids. Furthermore, we have performed an assay to ascertain tubular functionality. Following, we have imaged organoids containing Wt1-GFP knock-in cells live on the light sheet microscope to evaluate the ability to follow cells long-term and generate previously unavailable quantitative tracking data.\\n\\nidr0038 consists of a set of 3 experiments: experimentA (immunofluorescence staining after end point fixation), experimentB (assays to ascertain tubular functionality) and experimentC (Wt1-GFP knock-in cells followed long term).',\n",
       "  'id': 354,\n",
       "  'name': 'idr0038-held-kidneylightsheet'},\n",
       " {'description': 'Publication Title\\nEx vivo live cell tracking in kidney organoids using light sheet fluorescence microscopy\\n\\nExperiment Description\\nAll assays were performed on live cultures. The intact kidney rudiments were cultured at air liquid interface for 4 days following dissection. The organoids were cultured under medium immersion for six days following dissection. The samples cultured at air liquid interface were gently removed from the membranes before starting the assay. The samples were incubated for one hour at 25oC with PBS containing 1uM 5(6)-Carboxyfluorescein (6-CF) and 20 ug/ml of PNA (Vector Laboratories). For the control samples, 2 mM probenecid was also added to the solution to inhibit the organic anion transport. After incubation, the samples were washed with ice cold PBS for 10 mins followed by an incubation with 8mM probenecid in PBS for 15 mins to arrest any transport. The functional assay was concluded by two PBS washes and samples were then embedded in 1.5% agarose for imaging.',\n",
       "  'id': 352,\n",
       "  'name': 'idr0038-held-kidneylightsheet/experimentB'},\n",
       " {'description': 'Publication Title\\nEx vivo live cell tracking in kidney organoids using light sheet fluorescence microscopy\\n\\nExperiment Description\\nWe have adapted the mouse kidney rudiment assay to generate renal organoids. To demonstrate that the organoid culture method of re-aggregated kidney rudiments in PDMS discs resulted in 3D kidney organoids with organotypic structures, we stained fixed six day old organoids for various developmental markers and imaged them with a light sheet fluorescent microscope. We have stained for the following markers: Megalin, Laminin, Cytokeratin, Pax2, Six2, Wt1, Synaptopodin and Nephrin and have also used the lectin stain PNA (peanut agglutinin). The spheroids were usually imaged as z stacks from five angles.',\n",
       "  'id': 351,\n",
       "  'name': 'idr0038-held-kidneylightsheet/experimentA'},\n",
       " {'description': 'Experiment Description\\nHistopathology raw images and annotated tiff files of tissues from mice with 10 different single gene knockouts.',\n",
       "  'id': 101,\n",
       "  'name': 'idr0018-neff-histopathology/experimentA'},\n",
       " {'description': 'Publication Title\\nSubdiffraction imaging of centrosomes reveals higher-order organizational features of pericentriolar material.\\n\\nExperiment Description\\nImages relating to Figure 1e in Lawo et al 2012. These are the 3D-SIM reconstructed, maximum intensity projected and aligned images of centriole or PCM proteins of cycling HeLa cells in interphase.',\n",
       "  'id': 51,\n",
       "  'name': 'idr0021-lawo-pericentriolarmaterial/experimentA'},\n",
       " {'description': 'Publication Title\\nNuclear pore scaffold structure analyzed by super-resolution microscopy and particle averaging.\\n\\nExperiment Description\\nSystematic immunolabelling of the Nup107-160 complex and (monomeric) enhanced GFP [(m)EGFP] fusions of members of the Nup107-160 complex with an anti-GFP nanobody.',\n",
       "  'id': 52,\n",
       "  'name': 'idr0023-szymborska-nuclearpore/experimentA'},\n",
       " {'description': 'Publication Title\\nRegulation of Meristem Morphogenesis by Cell Wall Synthases in Arabidopsis.\\n\\nExperiment Description\\nWe performed systematic RNA in situ hybridization assays to examine the expression patterns of the SAM-expressed GTs at single-cell spatial resolution, including those transcripts that show either relative increases, reductions, or no change between SAM and flower samples.',\n",
       "  'id': 201,\n",
       "  'name': 'idr0032-yang-meristem/experimentA'},\n",
       " {'description': 'Publication Title\\nFocusing and sustaining the antitumor CTL effector killer response by agonist anti-CD137 mAb.\\n\\nExperiment Description\\nCombination therapy of adoptive CTL transfer and anti-CD137 monoclonal antibody. Mice were injected with B16F10/OVA melanoma cells and after 3 or 7 days were treated with OVA-specific CD8+ T cells (Cytotoxic T lymphocytes CTL) plus 1D8 antibody (anti-CD137).  4D timelapse images were taken at either 3 or 6 days after CTL + antibody treatment.\\n\\nVersion History\\nAugust 2017 - added additional phenotype to CMPO ontology mapping',\n",
       "  'id': 301,\n",
       "  'name': 'idr0026-weigelin-immunotherapy/experimentA'}]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "idr_project_list = get_all_projects(IDR_URL)\n",
    "idr_project_list"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get all the project attributes from IDR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "https://idr.openmicroscopy.org/webclient/api/annotations/?type=map&project=701\n",
      "https://idr.openmicroscopy.org/webclient/api/annotations/?type=map&project=505\n",
      "https://idr.openmicroscopy.org/webclient/api/annotations/?type=map&project=503\n",
      "https://idr.openmicroscopy.org/webclient/api/annotations/?type=map&project=504\n",
      "https://idr.openmicroscopy.org/webclient/api/annotations/?type=map&project=502\n",
      "https://idr.openmicroscopy.org/webclient/api/annotations/?type=map&project=552\n",
      "https://idr.openmicroscopy.org/webclient/api/annotations/?type=map&project=501\n",
      "https://idr.openmicroscopy.org/webclient/api/annotations/?type=map&project=405\n",
      "https://idr.openmicroscopy.org/webclient/api/annotations/?type=map&project=353\n",
      "https://idr.openmicroscopy.org/webclient/api/annotations/?type=map&project=151\n",
      "https://idr.openmicroscopy.org/webclient/api/annotations/?type=map&project=401\n",
      "https://idr.openmicroscopy.org/webclient/api/annotations/?type=map&project=404\n",
      "https://idr.openmicroscopy.org/webclient/api/annotations/?type=map&project=402\n",
      "https://idr.openmicroscopy.org/webclient/api/annotations/?type=map&project=354\n",
      "https://idr.openmicroscopy.org/webclient/api/annotations/?type=map&project=352\n",
      "https://idr.openmicroscopy.org/webclient/api/annotations/?type=map&project=351\n",
      "https://idr.openmicroscopy.org/webclient/api/annotations/?type=map&project=101\n",
      "https://idr.openmicroscopy.org/webclient/api/annotations/?type=map&project=51\n",
      "https://idr.openmicroscopy.org/webclient/api/annotations/?type=map&project=52\n",
      "https://idr.openmicroscopy.org/webclient/api/annotations/?type=map&project=201\n",
      "https://idr.openmicroscopy.org/webclient/api/annotations/?type=map&project=301\n"
     ]
    }
   ],
   "source": [
    "idr_attrib = get_all_proj_attrib(IDR_URL, idr_project_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'description': 'Publication Title\\nHuman lymphoid organ cDC2 and macrophages play complementary roles in the induction of T follicular helper responses\\n\\nExperiment Description\\nImaging mass cytometry of tonsil sections',\n",
       "  'id': 701,\n",
       "  'name': 'idr0054-segura-tonsilhyperion/experimentA',\n",
       "  'Study Type': 'imaging mass cytometry',\n",
       "  'Organism': 'Homo sapiens',\n",
       "  'Imaging Method': 'imaging mass cytometry',\n",
       "  'Publication Title': 'Human lymphoid organ cDC2 and macrophages play complementary roles in the induction of T follicular helper responses',\n",
       "  'Publication Authors': 'Mélanie Durand, Thomas Walter, Tiphène Pirnay, Thomas Naessens, Paul Gueguen, Christel Goudot, Sonia Lameiras, Qing Chang, Nafiseh Talaei, Olga Ornatsky, Tatiana Vassilevskaia, Sylvain Baulande, Sebastian Amigorena, Elodie Segura',\n",
       "  'License': 'CC-BY 4.0 https://creativecommons.org/licenses/by/4.0/',\n",
       "  'Copyright': 'Durand et al.',\n",
       "  'Data Publisher': 'University of Dundee',\n",
       "  'BioStudies Accession': 'S-BSST221 https://www.ebi.ac.uk/biostudies/studies/S-BSST221',\n",
       "  'Annotation File': 'idr0054-experimentA-annotation.csv https://github.com/IDR/idr0054-segura-tonsilhyperion/blob/master/experimentA/idr0054-experimentA-annotation.csv'},\n",
       " {'description': 'Publication Title\\nUnbiased pattern analysis reveals highly diverse responses of cytoskeletal systems to cyclic straining\\n\\nExperiment Description\\nCharacterization of cytoskeletal systems (actin, microtubules, vimentin networks) to cyclic straining using confocal laser scanning microscopy, including experiments with nocodazole to depolymerize microtubules.',\n",
       "  'id': 505,\n",
       "  'name': 'idr0050-springer-cytoskeletalsystems/experimentA',\n",
       "  'Study Type': 'Confocal laser scanning microscopy of cytoskeletal systems',\n",
       "  'Organism': 'Homo sapiens',\n",
       "  'Imaging Method': 'fluorescence microscopy',\n",
       "  'Publication Title': 'Unbiased pattern analysis reveals highly diverse responses of cytoskeletal systems to cyclic straining',\n",
       "  'Publication Authors': 'Ronald Springer, Alexander Zielinski, Catharina Pleschka, Bernd Hoffmann, Rudolf Merkel',\n",
       "  'PubMed ID': '30865622 https://www.ncbi.nlm.nih.gov/pubmed/30865622',\n",
       "  'Publication DOI': '10.1371/journal.pone.0210570 https://doi.org/10.1371/journal.pone.0210570',\n",
       "  'License': 'CC-BY 4.0 https://creativecommons.org/licenses/by/4.0/',\n",
       "  'Copyright': 'Springer et al',\n",
       "  'Data Publisher': 'University of Dundee',\n",
       "  'Data DOI': '10.17867/10000119 https://doi.org/10.17867/10000119',\n",
       "  'Annotation File': 'idr0050-experimentA-annotation.csv https://github.com/IDR/idr0050-springer-cytoskeletalsystems/blob/master/experimentA/idr0050-experimentA-annotation.csv'},\n",
       " {'description': 'Publication Title\\nA microscopy data set of discrete spatial and temporal single molecule RNA expression in single cells\\n\\nExperiment Description\\nWe performed single molecule in-situ hybridization on yeast cells that have been exposed to osmotic stress and measured mRNA expression for two genes. The goal of this study was to generate data sets for single cell transcription modeling.',\n",
       "  'id': 503,\n",
       "  'name': 'idr0047-neuert-yeastmrna/experimentA',\n",
       "  'Study Type': '',\n",
       "  'Organism': 'Saccharomyces cerevisiae',\n",
       "  'Imaging Method': 'fluorescent in situ hybridization',\n",
       "  'Publication Title': 'A microscopy data set of discrete spatial and temporal single molecule RNA expression in single cells',\n",
       "  'Publication Authors': 'Guoliang Li, Gregor Neuert',\n",
       "  'License': 'CC-BY 4.0 https://creativecommons.org/licenses/by/4.0/',\n",
       "  'Copyright': 'Neuert et al.',\n",
       "  'Data Publisher': 'University of Dundee',\n",
       "  'Data DOI': '10.17867/10000118 https://doi.org/10.17867/10000118',\n",
       "  'BioStudies Accession': 'S-BIAD1 https://www.ebi.ac.uk/biostudies/studies/S-BIAD1',\n",
       "  'Annotation File': 'idr0047-experimentA-annotation.csv https://github.com/IDR/idr0047-neuert-yeastmrna/blob/master/experimentA/idr0047-experimentA-annotation.csv'},\n",
       " {'description': 'Publication Title\\nVirtual nanoscopy: Generation of ultra-large high resolution electron microscopy maps\\n\\nExperiment Description\\n\\tUltra-large high resolution electron microscopy maps of the mouse glomerulus, mouse embryonic fibroblasts, human dendritic cells and a zebrafish embryo slice were generated using virtual nanoscopy.',\n",
       "  'id': 504,\n",
       "  'name': 'idr0053-faas-virtualnanoscopy/experimentA',\n",
       "  'Study Type': 'high resolution electron microscopy',\n",
       "  'Organism': 'Homo sapiens',\n",
       "  'Imaging Method': 'Transmission electron microscopy',\n",
       "  'Publication Title': 'Virtual nanoscopy: Generation of ultra-large high resolution electron microscopy maps',\n",
       "  'Publication Authors': 'Frank G.A. Faas, M. Cristina Avramut, Bernard M. van den Berg, A. Mieke Mommaas, Abraham J. Koster, Raimond B.G. Ravelli',\n",
       "  'PubMed ID': '22869601 https://www.ncbi.nlm.nih.gov/pubmed/22869601',\n",
       "  'PMC ID': 'PMC3413355 https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3413355',\n",
       "  'Publication DOI': '10.1083/jcb.201201140 https://doi.org/10.1083/jcb.201201140',\n",
       "  'License': 'CC-BY-NC-SA https://creativecommons.org/licenses/by-nc-sa/3.0/',\n",
       "  'Copyright': 'Faas et al.',\n",
       "  'Data Publisher': 'University of Dundee',\n",
       "  'Data DOI': '10.1083/jcb.201201140.dv https://doi.org/10.1083/jcb.201201140.dv',\n",
       "  'Annotation File': 'idr0053-experimentA-annotation.csv https://github.com/IDR/idr0053-faas-virtualnanoscopy/blob/master/experimentA/idr0053-experimentA-annotation.csv'},\n",
       " {'description': 'Publication Title\\nIn Toto Imaging and Reconstruction of Post-Implantation Mouse Development at the Single-Cell Level\\n\\nExperiment Description\\nLight-sheet imaging of post-implantation mouse development from E6.5 to E8.5',\n",
       "  'id': 502,\n",
       "  'name': 'idr0044-mcdole-tardislightsheet/experimentA',\n",
       "  'Study Type': 'light sheet fluorescence microscopy of TARDIS registered embryos',\n",
       "  'Organism': 'Mus musculus',\n",
       "  'Imaging Method': 'Light-sheet fluorescence microscopy',\n",
       "  'Publication Title': 'In Toto Imaging and Reconstruction of Post-Implantation Mouse Development at the Single-Cell Level',\n",
       "  'Publication Authors': 'Katie McDole, Léo Guignard, Fernando Amat, Andrew Berger, Grégoire Malandain, Loïc A. Royer, Srinivas C. Turaga, Kristin Branson, Philipp J. Keller',\n",
       "  'PubMed ID': '30318151',\n",
       "  'Publication DOI': 'https://doi.org/10.1016/j.cell.2018.09.031',\n",
       "  'License': 'CC-BY 4.0 https://creativecommons.org/licenses/by/4.0/',\n",
       "  'Copyright': 'McDole et al.',\n",
       "  'Data Publisher': 'University of Dundee',\n",
       "  'Data DOI': '10.17867/10000116 https://doi.org/10.17867/10000116',\n",
       "  'Annotation File': 'idr0044-experimentA-annotation.csv https://github.com/IDR/idr0044-mcdole-tardislightsheet/blob/master/experimentA/idr0044-experimentA-annotation.csv'},\n",
       " {'description': 'Publication Title\\nNeuromesodermal Progenitors are a Conserved Source of Spinal Cord with Divergent Growth Dynamics\\n\\nExperiment Description\\nLightsheet imaging of tailbud from 18 somite stage with one frame per 2 minutes',\n",
       "  'id': 552,\n",
       "  'name': 'idr0051-fulton-tailbudlightsheet/experimentA',\n",
       "  'Study Type': 'light sheet microscopy of zebrafish tailbud',\n",
       "  'Organism': 'Danio rerio',\n",
       "  'Imaging Method': 'SPIM',\n",
       "  'Publication Title': 'Neuromesodermal Progenitors are a Conserved Source of Spinal Cord with Divergent Growth Dynamics',\n",
       "  'Publication Authors': 'Andrea Attardi, Timothy Fulton, Maria Florescu, Gopi Shah, Leila Muresan, Martin O. Lenz, Courtney Lancaster, Jan Huisken, Alexander van Oudenaarden and Benjamin Steventon',\n",
       "  'PubMed ID': '30333213 https://www.ncbi.nlm.nih.gov/pubmed/30333213',\n",
       "  'PMC ID': 'PMC6240315 https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6240315',\n",
       "  'Publication DOI': '10.1242/dev.166728 https://doi.org/10.1242/dev.166728',\n",
       "  'License': 'CC-BY 4.0 https://creativecommons.org/licenses/by/4.0/',\n",
       "  'Copyright': 'Steventon et al.',\n",
       "  'Data Publisher': 'University of Dundee',\n",
       "  'Data DOI': '10.17867/10000117 https://doi.org/10.17867/10000117',\n",
       "  'Annotation File': 'idr0051-experimentA-annotation.csv https://github.com/IDR/idr0051-fulton-tailbudlightsheet/blob/master/experimentA/idr0051-experimentA-annotation.csv'},\n",
       " {'description': 'Publication Title\\nTissue-based map of the human proteome.\\n\\nExperiment Description\\nMapping of protein expression across all major tissues and organs in the human body',\n",
       "  'id': 501,\n",
       "  'name': 'idr0043-uhlen-humanproteinatlas/experimentA',\n",
       "  'Study Type': 'protein localization',\n",
       "  'Organism': 'Homo sapiens',\n",
       "  'Imaging Method': 'bright-field microscopy',\n",
       "  'Publication Title': 'Tissue-based map of the human proteome',\n",
       "  'Publication Authors': 'Uhlen M et al',\n",
       "  'PubMed ID': '25613900',\n",
       "  'Publication DOI': 'http://dx.doi.org/10.1126/science.1260419',\n",
       "  'License': 'CC BY-SA 3.0 https://v18.proteinatlas.org/about/licence',\n",
       "  'Copyright': 'Uhlen M et al',\n",
       "  'Data Publisher': 'University of Dundee',\n",
       "  'Data DOI': '---',\n",
       "  'Annotation File': 'idr0043-experimentA-annotation.csv.gz https://github.com/IDR/idr0043-uhlen-humanproteinatlas/blob/master/experimentA/hpa_run_01/idr0043-experimentA-annotation.csv.gz'},\n",
       " {'description': 'Publication Title\\nDual spindle formation in zygotes keeps parental genomes apart in early mammalian embryos\\n\\nExperiment Description\\nCharacterization of mitotic spindle formation in live mouse zygotes by light-sheet microscopy.',\n",
       "  'id': 405,\n",
       "  'name': 'idr0045-reichmann-zygotespindle/experimentA',\n",
       "  'Study Type': '5D light sheet and confocal microscopy imaging of mouse zygotes',\n",
       "  'Organism': 'Mus musculus x Mus spretus',\n",
       "  'Imaging Method': 'confocal microscopy',\n",
       "  'Publication Title': 'Dual spindle formation in zygotes keeps parental genomes apart in early mammalian embryos',\n",
       "  'Publication Authors': 'Judith Reichmann, Bianca Nijmeijer, M. Julius Hossain, Manuel Eguren, Isabell Schneider, Antonio Z. Politi, Maria J. Roberti, Lars Hufnagel, Takashi Hiiragi and Jan Ellenberg',\n",
       "  'PubMed ID': '30002254 https://www.ncbi.nlm.nih.gov/pubmed/30002254',\n",
       "  'Publication DOI': '10.1126/science.aar7462 https://doi.org/10.1126/science.aar7462',\n",
       "  'License': 'CC-BY 4.0 https://creativecommons.org/licenses/by/4.0/',\n",
       "  'Copyright': 'Reichmann et al',\n",
       "  'Data Publisher': 'University of Dundee',\n",
       "  'Data DOI': '10.17867/10000115 https://doi.org/10.17867/10000115',\n",
       "  'Annotation File': 'idr0045-experimentA-annotation.csv https://github.com/IDR/idr0045-reichmann-zygotespindle/blob/master/experimentA/idr0045-experimentA-annotation.csv'},\n",
       " {'description': 'Publication Title\\nEx vivo live cell tracking in kidney organoids using light sheet fluorescence microscopy\\n\\nExperiment Description\\nWe have adapted the mouse kidney rudiment assay to generate renal organoids. 24 hours after pelleting 100,000 mouse kidney cells, organoids had formed in the PDMS dishes. For the PNA vital stain test we used kidney cells from wildtype mice and for the tracking time series we used cells from the Wt1tm1Nhsn strain, expressing GFP together with the transcription factor Wt1. One organoid was embedded in a hydrogel matrix (1.5% Agarose + 3% gelatine). In the microscope chamber, the hydrogel cylinder was extruded from the capillary so that the hydrogel column containing the organoid was located outside the capillary and immersed in medium (phenol red-free MEM + 10% FBS, 1% pen/strep, 3ug/ml PNA-rh) at 37oC and with 5% CO2. We recorded z stacks of the spheroids from multiple angles every 30 mins to record the location of the GFP expressing cells in three dimensions over time. There are three time series. These are available in the subfolders \"Time PNA-rh vital\", \"Wt1-GFP PNA Time Series 1\" and \"Wt1-GFP PNA Time Series 2\".  The time series data was processed in Fiji by opening the green channel only, binning in X and Y by a factor of two and converting it to 8 bit for processing in TrackMate (v.2.8.2) where we used the LoG (Laplacian of Gaussian) Detector and LAP (Linear Assignment Problem) Tracker options. Tracking data generated from the time series using the Track Mate plug in for Fiji is also provided in the time series data sets as \"Nspots vs Time.txt\", \"Track statistics.txt\", \"Links in tracks statistics.txt\" and \"Spots in tracks statistics.txt\". The data was analysed and plotted via the Matlab script provided as \"TrackMateDataAnalysis.txt\" in conjunction with the functions \"NSpotsFunction.txt\", \"SpotsInTracksFunction.txt\", \"TrackStatisticsFunction.txt\", \"Violin.txt\" and \"violinplot.txt\".',\n",
       "  'id': 353,\n",
       "  'name': 'idr0038-held-kidneylightsheet/experimentC',\n",
       "  'Study Type': 'light sheet fluorescence microscopy of organoids',\n",
       "  'Organism': 'Mus musculus',\n",
       "  'Imaging Method': 'light sheet fluorescence microscopy',\n",
       "  'Publication Title': 'Ex vivo live cell tracking in kidney organoids using light sheet fluorescence microscopy',\n",
       "  'Publication Authors': 'Marie Held, Ilaria Santeramo, Bettina Wilm, Patricia Murray, Raphael Levy',\n",
       "  'PubMed ID': '30048451 https://www.ncbi.nlm.nih.gov/pubmed/30048451',\n",
       "  'PMC ID': 'PMC6062017 https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6062017',\n",
       "  'Publication DOI': '10.1371/journal.pone.0199918 https://doi.org/10.1371/journal.pone.0199918',\n",
       "  'License': 'CC-BY 4.0 https://creativecommons.org/licenses/by/4.0/',\n",
       "  'Copyright': 'Held et al',\n",
       "  'Data Publisher': 'University of Dundee',\n",
       "  'Data DOI': '10.17867/10000110c https://doi.org/10.17867/10000110c',\n",
       "  'Study': 'Overview https://idr.openmicroscopy.org/webclient/?show=project-354',\n",
       "  'Annotation File': 'idr0038-experimentC-annotation.csv https://github.com/IDR/idr-metadata/blob/master/idr0038-held-kidneylightsheet/experimentC/idr0038-experimentC-annotation.csv'},\n",
       " {'description': 'Publication Title\\nHigh resolution imaging reveals heterogeneity in chromatin states between cells that is not inherited through cell division.\\n\\nExperiment Description\\nTo monitor the organisation of chromatin in live cells were engineered with fluorescent fusion proteins which recognize specific operator sequences to tag pairs of syntenic gene loci. The separation of these loci was then tracked in three dimensions over time using fluorescence microscopy. Data generated for this manuscript are made accessible here. This includes the movies generated from yeast strains with different genomic separations. These are available within subfolders titled \"Colocalising\", \"Genomic separation 100kb\", \"Genomic separation 25kb\" etc. Tracking data generated from these movies is also available as idr0027-AnalysisAllData.csv in the Attachments section below. This includes x, y and z co-ordinates of foci in the red and green channels following two stages of correction for channel alignment, the distance between the red and green foci in nanometers,  the offset of the red channel foci with respect to the green channel foci in x, y, and z. The maximum intensity, background intensity and standard deviation of the background intensity for both the red and green channels. The contrast (maximum spot intensity - nuclear background intensity) for each channel. The name originally assigned to the video, the genomic separation of the strain in kb. The string in column AE starts with the letter assigned to the video in supplementary figure 2. Some videos include two cells. In these cases the higher cell (on the y axis) is indicated as \"upper\" and the other as \"lower\". One of the two cells can often be assigned as the mother and the other as daughter. This is indicated as m or d in column AF. The systematic name of the strain is indicated in column AG.',\n",
       "  'id': 151,\n",
       "  'name': 'idr0027-dickerson-chromatin/experimentA',\n",
       "  'Study Type': '3D-tracking of tagged chromatin loci',\n",
       "  'Organism': 'Saccharomyces cerevisiae',\n",
       "  'Imaging Method': 'fluorescence microscopy',\n",
       "  'Publication Title': 'High resolution imaging reveals heterogeneity in chromatin states between cells that is not inherited through cell division.',\n",
       "  'Publication Authors': 'Dickerson D, Gierliski M, Singh V, Kitamura E, Ball G, Tanaka TU and Owen-Hughes T',\n",
       "  'PubMed ID': '27609610 https://www.ncbi.nlm.nih.gov/pubmed/27609610',\n",
       "  'PMC ID': 'PMC5016949 https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5016949',\n",
       "  'Publication DOI': '10.1186/s12860-016-0111-y https://doi.org/10.1186/s12860-016-0111-y',\n",
       "  'License': 'CC-BY 4.0 https://creativecommons.org/licenses/by/4.0/',\n",
       "  'Copyright': 'Dickerson et al',\n",
       "  'Data Publisher': 'University of Dundee',\n",
       "  'Data DOI': '10.17867/10000102 https://doi.org/10.17867/10000102',\n",
       "  'Annotation File': 'idr0027-experimentA-annotation.csv https://github.com/IDR/idr-metadata/blob/master/idr0027-dickerson-chromatin/experimentA/idr0027-experimentA-annotation.csv'},\n",
       " {'description': 'Publication Title\\nTiming of gene expression in a cell-fate decision system\\n\\nExperiment Description\\nTime-lapse with addition of 1uM alpha-factor at time 0',\n",
       "  'id': 401,\n",
       "  'name': 'idr0040-aymoz-singlecell/experimentA',\n",
       "  'Study Type': 'timelapse microscopy',\n",
       "  'Organism': 'Saccharomyces cerevisiae',\n",
       "  'Imaging Method': 'fluorescent microscopy',\n",
       "  'Publication Title': 'Timing of gene expression in a cell-fate decision system',\n",
       "  'Publication Authors': 'Delphine Aymoz, Carme Solé, Jean-Jerrold Pierre, Marta Schmitt, Eulàlia de Nadal, Francesc Posas, Serge Pelet',\n",
       "  'PubMed ID': '29695607 https://www.ncbi.nlm.nih.gov/pubmed/29695607',\n",
       "  'PMC ID': 'PMC5916086 https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5916086',\n",
       "  'Publication DOI': '10.15252/msb.20178024 https://doi.org/10.15252/msb.20178024',\n",
       "  'License': 'CC-BY 4.0 https://creativecommons.org/licenses/by/4.0/',\n",
       "  'Copyright': 'Pelet et al',\n",
       "  'Data Publisher': 'University of Dundee',\n",
       "  'Data DOI': '10.17867/10000114 https://doi.org/10.17867/10000114',\n",
       "  'Annotation File': 'idr0040-experimentA-annotation.csv https://github.com/IDR/idr0040-aymoz-singlecell/blob/master/experimentA/idr0040-experimentA-annotation.csv'},\n",
       " {'description': 'Publication Title\\nExperimental and computational framework for a dynamic protein atlas of human cell division\\n\\nExperiment Description\\nImage data used to build the mitotic cell atlas. This consists of three subsets: raw images, segmentation masks for the landmarks and protein concentration maps.',\n",
       "  'id': 404,\n",
       "  'name': 'idr0041-cai-mitoticatlas/experimentA',\n",
       "  'Study Type': 'Quantitative protein localization using FCS-calibrated 3D time-lapse imaging',\n",
       "  'Organism': 'Homo sapiens',\n",
       "  'Imaging Method': 'fluorescence correlation spectroscopy',\n",
       "  'Publication Title': 'Experimental and computational framework for a dynamic protein atlas of human cell division',\n",
       "  'Publication Authors': 'Yin Cai, M. Julius Hossain, Jean-Karim Hériché, Antonio Z. Politi, Nike Walther, Birgit Koch, Malte Wachsmuth, Bianca Nijmeijer, Moritz Kueblbeck, Marina Martinic Kavur, Rene Ladurner, Jan-Michael Peters & Jan Ellenberg',\n",
       "  'PubMed ID': '30202089 https://www.ncbi.nlm.nih.gov/pubmed/30202089',\n",
       "  'Publication DOI': '10.1038/s41586-018-0518-z https://doi.org/10.1038/s41586-018-0518-z',\n",
       "  'License': 'CC-BY 4.0 https://creativecommons.org/licenses/by/4.0/',\n",
       "  'Copyright': 'Cai et al',\n",
       "  'Data Publisher': 'University of Dundee',\n",
       "  'Data DOI': '10.17867/10000112 https://doi.org/10.17867/10000112',\n",
       "  'Annotation File': 'idr0041-experimentA-annotation.csv https://github.com/IDR/idr0041-cai-mitoticatlas/blob/master/experimentA/idr0041-experimentA-annotation.csv'},\n",
       " {'description': 'Publication Title\\nA Deep-learning classifier identifies patients with clinical heart failure using whole-slide images of H&E tissue\\n\\nExperiment Description\\nHistopathology sub-image from patient whole-slide images from patients with end-stage clinical heart failure or cadaveric donor hearts from patients without heart failure.',\n",
       "  'id': 402,\n",
       "  'name': 'idr0042-nirschl-wsideeplearning/experimentA',\n",
       "  'Study Type': 'histology',\n",
       "  'Organism': 'Homo sapiens',\n",
       "  'Imaging Method': 'bright-field microscopy',\n",
       "  'Publication Title': 'A Deep-learning classifier identifies patients with clinical heart failure using whole-slide images of H&E tissue',\n",
       "  'Publication Authors': 'Jeffrey J. Nirschl, Andrew Janowczyk, Eliot G. Peyster, Renee Frank, Kenneth B. Margulies, Michael D. Feldman, Anant Madabhushi',\n",
       "  'PubMed ID': '29614076 https://www.ncbi.nlm.nih.gov/pubmed/29614076',\n",
       "  'PMC ID': 'PMC5882098 https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5882098',\n",
       "  'Publication DOI': '10.1371/journal.pone.0192726 https://doi.org/10.1371/journal.pone.0192726',\n",
       "  'License': 'CC-BY 4.0 https://creativecommons.org/licenses/by/4.0/',\n",
       "  'Copyright': 'Nirschl et al',\n",
       "  'Data Publisher': 'University of Dundee',\n",
       "  'Data DOI': '10.17867/10000113 https://doi.org/10.17867/10000113',\n",
       "  'Annotation File': 'idr0042-experimentA-annotation.csv https://github.com/IDR/idr0042-nirschl-wsideeplearning/blob/master/experimentA/idr0042-experimentA-annotation.csv'},\n",
       " {'description': 'Publication Title\\nEx vivo live cell tracking in kidney organoids using light sheet fluorescence microscopy\\n\\nStudy Description\\nWe have adapted the mouse kidney rudiment assay to generate renal organoids. We have immunostained endpoint fixed organoids and E13.5 embryonic kidneys and imaged them with a light sheet fluorescent microscope. The data generated for this manuscript is made accessible here and includes immunofluorescence staining after end point fixation revealing that organotypic structures develop within the organoids. Furthermore, we have performed an assay to ascertain tubular functionality. Following, we have imaged organoids containing Wt1-GFP knock-in cells live on the light sheet microscope to evaluate the ability to follow cells long-term and generate previously unavailable quantitative tracking data.\\n\\nidr0038 consists of a set of 3 experiments: experimentA (immunofluorescence staining after end point fixation), experimentB (assays to ascertain tubular functionality) and experimentC (Wt1-GFP knock-in cells followed long term).',\n",
       "  'id': 354,\n",
       "  'name': 'idr0038-held-kidneylightsheet',\n",
       "  'Study Type': 'light sheet fluorescence microscopy of organoids',\n",
       "  'Publication Title': 'Ex vivo live cell tracking in kidney organoids using light sheet fluorescence microscopy',\n",
       "  'Publication Authors': 'Marie Held, Ilaria Santeramo, Bettina Wilm, Patricia Murray, Raphael Levy',\n",
       "  'PubMed ID': '30048451 https://www.ncbi.nlm.nih.gov/pubmed/30048451',\n",
       "  'PMC ID': 'PMC6062017 https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6062017',\n",
       "  'Publication DOI': '10.1371/journal.pone.0199918 https://doi.org/10.1371/journal.pone.0199918',\n",
       "  'License': 'CC-BY 4.0 https://creativecommons.org/licenses/by/4.0/',\n",
       "  'Copyright': 'Held et al',\n",
       "  'Data Publisher': 'University of Dundee',\n",
       "  'Data DOI': '10.17867/10000110 https://doi.org/10.17867/10000110',\n",
       "  'Study': 'Experiment C https://idr.openmicroscopy.org/webclient/?show=project-353'},\n",
       " {'description': 'Publication Title\\nEx vivo live cell tracking in kidney organoids using light sheet fluorescence microscopy\\n\\nExperiment Description\\nAll assays were performed on live cultures. The intact kidney rudiments were cultured at air liquid interface for 4 days following dissection. The organoids were cultured under medium immersion for six days following dissection. The samples cultured at air liquid interface were gently removed from the membranes before starting the assay. The samples were incubated for one hour at 25oC with PBS containing 1uM 5(6)-Carboxyfluorescein (6-CF) and 20 ug/ml of PNA (Vector Laboratories). For the control samples, 2 mM probenecid was also added to the solution to inhibit the organic anion transport. After incubation, the samples were washed with ice cold PBS for 10 mins followed by an incubation with 8mM probenecid in PBS for 15 mins to arrest any transport. The functional assay was concluded by two PBS washes and samples were then embedded in 1.5% agarose for imaging.',\n",
       "  'id': 352,\n",
       "  'name': 'idr0038-held-kidneylightsheet/experimentB',\n",
       "  'Study Type': 'light sheet fluorescence microscopy of organoids',\n",
       "  'Organism': 'Mus musculus',\n",
       "  'Imaging Method': 'light sheet fluorescence microscopy',\n",
       "  'Publication Title': 'Ex vivo live cell tracking in kidney organoids using light sheet fluorescence microscopy',\n",
       "  'Publication Authors': 'Marie Held, Ilaria Santeramo, Bettina Wilm, Patricia Murray, Raphael Levy',\n",
       "  'PubMed ID': '30048451 https://www.ncbi.nlm.nih.gov/pubmed/30048451',\n",
       "  'PMC ID': 'PMC6062017 https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6062017',\n",
       "  'Publication DOI': '10.1371/journal.pone.0199918 https://doi.org/10.1371/journal.pone.0199918',\n",
       "  'License': 'CC-BY 4.0 https://creativecommons.org/licenses/by/4.0/',\n",
       "  'Copyright': 'Held et al',\n",
       "  'Data Publisher': 'University of Dundee',\n",
       "  'Data DOI': '10.17867/10000110b https://doi.org/10.17867/10000110b',\n",
       "  'Study': 'Overview https://idr.openmicroscopy.org/webclient/?show=project-354',\n",
       "  'Annotation File': 'idr0038-experimentB-annotation.csv https://github.com/IDR/idr-metadata/blob/master/idr0038-held-kidneylightsheet/experimentB/idr0038-experimentB-annotation.csv'},\n",
       " {'description': 'Publication Title\\nEx vivo live cell tracking in kidney organoids using light sheet fluorescence microscopy\\n\\nExperiment Description\\nWe have adapted the mouse kidney rudiment assay to generate renal organoids. To demonstrate that the organoid culture method of re-aggregated kidney rudiments in PDMS discs resulted in 3D kidney organoids with organotypic structures, we stained fixed six day old organoids for various developmental markers and imaged them with a light sheet fluorescent microscope. We have stained for the following markers: Megalin, Laminin, Cytokeratin, Pax2, Six2, Wt1, Synaptopodin and Nephrin and have also used the lectin stain PNA (peanut agglutinin). The spheroids were usually imaged as z stacks from five angles.',\n",
       "  'id': 351,\n",
       "  'name': 'idr0038-held-kidneylightsheet/experimentA',\n",
       "  'Study Type': 'light sheet fluorescence microscopy of organoids',\n",
       "  'Organism': 'Mus musculus',\n",
       "  'Imaging Method': 'light sheet fluorescence microscopy',\n",
       "  'Publication Title': 'Ex vivo live cell tracking in kidney organoids using light sheet fluorescence microscopy',\n",
       "  'Publication Authors': 'Marie Held, Ilaria Santeramo, Bettina Wilm, Patricia Murray, Raphael Levy',\n",
       "  'PubMed ID': '30048451 https://www.ncbi.nlm.nih.gov/pubmed/30048451',\n",
       "  'PMC ID': 'PMC6062017 https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6062017',\n",
       "  'Publication DOI': '10.1371/journal.pone.0199918 https://doi.org/10.1371/journal.pone.0199918',\n",
       "  'License': 'CC-BY 4.0 https://creativecommons.org/licenses/by/4.0/',\n",
       "  'Copyright': 'Held et al',\n",
       "  'Data Publisher': 'University of Dundee',\n",
       "  'Data DOI': '10.17867/10000110a https://doi.org/10.17867/10000110a',\n",
       "  'Study': 'Overview https://idr.openmicroscopy.org/webclient/?show=project-354',\n",
       "  'Annotation File': 'idr0038-experimentA-annotation.csv https://github.com/IDR/idr-metadata/blob/master/idr0038-held-kidneylightsheet/experimentA/idr0038-experimentA-annotation.csv'},\n",
       " {'description': 'Experiment Description\\nHistopathology raw images and annotated tiff files of tissues from mice with 10 different single gene knockouts.',\n",
       "  'id': 101,\n",
       "  'name': 'idr0018-neff-histopathology/experimentA',\n",
       "  'Study Type': 'histology',\n",
       "  'Organism': 'Mus musculus',\n",
       "  'Imaging Method': 'bright-field microscopy',\n",
       "  'License': 'CC-BY 4.0 https://creativecommons.org/licenses/by/4.0/',\n",
       "  'Copyright': 'Frauke Neff and Jean-Karim Heriche',\n",
       "  'Annotation File': 'idr0018-experimentA-annotation.csv https://github.com/IDR/idr-metadata/blob/master/idr0018-neff-histopathology/experimentA/idr0018-experimentA-annotation.csv'},\n",
       " {'description': 'Publication Title\\nSubdiffraction imaging of centrosomes reveals higher-order organizational features of pericentriolar material.\\n\\nExperiment Description\\nImages relating to Figure 1e in Lawo et al 2012. These are the 3D-SIM reconstructed, maximum intensity projected and aligned images of centriole or PCM proteins of cycling HeLa cells in interphase.',\n",
       "  'id': 51,\n",
       "  'name': 'idr0021-lawo-pericentriolarmaterial/experimentA',\n",
       "  'Study Type': 'protein localization using 3D-SIM',\n",
       "  'Organism': 'Homo sapiens',\n",
       "  'Imaging Method': 'structured illumination microscopy (SIM)',\n",
       "  'Publication Title': 'Subdiffraction imaging of centrosomes reveals higher-order organizational features of pericentriolar material.',\n",
       "  'Publication Authors': 'Lawo S, Hasegan M, Gupta GD, Pelletier L.',\n",
       "  'PubMed ID': '23086237 https://www.ncbi.nlm.nih.gov/pubmed/23086237',\n",
       "  'Publication DOI': '10.1038/ncb2591 https://doi.org/10.1038/ncb2591',\n",
       "  'License': 'CC-BY 4.0 https://creativecommons.org/licenses/by/4.0/',\n",
       "  'Copyright': 'Lawo et al',\n",
       "  'Annotation File': 'idr0021-experimentA-annotation.csv https://github.com/IDR/idr-metadata/blob/master/idr0021-lawo-pericentriolarmaterial/experimentA/idr0021-experimentA-annotation.csv'},\n",
       " {'description': 'Publication Title\\nNuclear pore scaffold structure analyzed by super-resolution microscopy and particle averaging.\\n\\nExperiment Description\\nSystematic immunolabelling of the Nup107-160 complex and (monomeric) enhanced GFP [(m)EGFP] fusions of members of the Nup107-160 complex with an anti-GFP nanobody.',\n",
       "  'id': 52,\n",
       "  'name': 'idr0023-szymborska-nuclearpore/experimentA',\n",
       "  'Study Type': 'protein localization using dSTORM',\n",
       "  'Organism': 'Homo sapiens',\n",
       "  'Imaging Method': 'STORM super-resolution imaging',\n",
       "  'Publication Title': 'Nuclear pore scaffold structure analyzed by super-resolution microscopy and particle averaging.',\n",
       "  'Publication Authors': 'Szymborska A, de Marco A, Daigle N, Cordes VC, Briggs JA, Ellenberg J.',\n",
       "  'PubMed ID': '23845946 https://www.ncbi.nlm.nih.gov/pubmed/23845946',\n",
       "  'Publication DOI': '10.1126/science.1240672 https://doi.org/10.1126/science.1240672',\n",
       "  'License': 'CC-BY 4.0 https://creativecommons.org/licenses/by/4.0/',\n",
       "  'Copyright': 'Szymborska et al',\n",
       "  'Annotation File': 'idr0023-experimentA-annotation.csv https://github.com/IDR/idr-metadata/blob/master/idr0023-szymborska-nuclearpore/experimentA/idr0023-experimentA-annotation.csv'},\n",
       " {'description': 'Publication Title\\nRegulation of Meristem Morphogenesis by Cell Wall Synthases in Arabidopsis.\\n\\nExperiment Description\\nWe performed systematic RNA in situ hybridization assays to examine the expression patterns of the SAM-expressed GTs at single-cell spatial resolution, including those transcripts that show either relative increases, reductions, or no change between SAM and flower samples.',\n",
       "  'id': 201,\n",
       "  'name': 'idr0032-yang-meristem/experimentA',\n",
       "  'Study Type': 'in situ hybridization',\n",
       "  'Organism': 'Arabidopsis thaliana',\n",
       "  'Imaging Method': 'fluorescence microscopy',\n",
       "  'Publication Title': 'Regulation of Meristem Morphogenesis by Cell Wall Synthases in Arabidopsis.',\n",
       "  'Publication Authors': 'Yang W, Schuster C, Beahan CT, Charoensawan V, Peaucelle A, Bacic A, Doblin MS, Wightman R, Meyerowitz EM',\n",
       "  'PubMed ID': '27212401 https://www.ncbi.nlm.nih.gov/pubmed/27212401',\n",
       "  'PMC ID': 'PMC5024349 https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5024349',\n",
       "  'Publication DOI': '10.1016/j.cub.2016.04.026 https://doi.org/10.1016/j.cub.2016.04.026',\n",
       "  'Annotation File': 'idr0032-experimentA-annotation.csv https://github.com/IDR/idr-metadata/blob/master/idr0032-yang-meristem/experimentA/idr0032-experimentA-annotation.csv'},\n",
       " {'description': 'Publication Title\\nFocusing and sustaining the antitumor CTL effector killer response by agonist anti-CD137 mAb.\\n\\nExperiment Description\\nCombination therapy of adoptive CTL transfer and anti-CD137 monoclonal antibody. Mice were injected with B16F10/OVA melanoma cells and after 3 or 7 days were treated with OVA-specific CD8+ T cells (Cytotoxic T lymphocytes CTL) plus 1D8 antibody (anti-CD137).  4D timelapse images were taken at either 3 or 6 days after CTL + antibody treatment.\\n\\nVersion History\\nAugust 2017 - added additional phenotype to CMPO ontology mapping',\n",
       "  'id': 301,\n",
       "  'name': 'idr0026-weigelin-immunotherapy/experimentA',\n",
       "  'Study Type': '4D timelapse of in vivo cells',\n",
       "  'Organism': 'Mus musculus',\n",
       "  'Imaging Method': 'multi-photon microscopy',\n",
       "  'Publication Title': 'Focusing and sustaining the antitumor CTL effector killer response by agonist anti-CD137 mAb.',\n",
       "  'Publication Authors': 'Weigelin B, Bolanos E, Teijeira A, Martinez-Forero I, Labiano S, Azpilikueta A, Morales-Kastresana A, Quetglas JI, Wagena E, Sanchez-Paulete AR, Chen L, Friedl P, Melero I.',\n",
       "  'PubMed ID': '26034288 https://www.ncbi.nlm.nih.gov/pubmed/26034288',\n",
       "  'PMC ID': 'PMC4475992 https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4475992',\n",
       "  'Publication DOI': '10.1073/pnas.1506357112 https://doi.org/10.1073/pnas.1506357112',\n",
       "  'License': 'CC-BY 4.0 https://creativecommons.org/licenses/by/4.0/',\n",
       "  'Copyright': 'Weigelin et al',\n",
       "  'Annotation File': 'idr0026-experimentA-annotation.csv https://github.com/IDR/idr-metadata/blob/master/idr0026-weigelin-immunotherapy/experimentA/idr0026-experimentA-annotation.csv'}]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "idr_attrib"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## searching IDR for the phrase \"mus\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'id': 502, 'Organism': 'Mus musculus'},\n",
       " {'id': 405, 'Organism': 'Mus musculus x Mus spretus'},\n",
       " {'id': 353, 'Organism': 'Mus musculus'},\n",
       " {'id': 352, 'Organism': 'Mus musculus'},\n",
       " {'id': 351, 'Organism': 'Mus musculus'},\n",
       " {'id': 101, 'Organism': 'Mus musculus'},\n",
       " {'id': 301, 'Organism': 'Mus musculus'}]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_proj(idr_attrib, 'mus')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## function combine_search() returns the projects that matches the search term in both SSBD-OMERO and IDR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "def combine_search(term):\n",
    "    idr_result = search_proj(idr_attrib, term)\n",
    "    ssbd_result = search_proj(ssbd_attrib, term)\n",
    "    print(\"IDR searh results:\")\n",
    "    for i in idr_result:\n",
    "        for key, value in i.items():\n",
    "            print(\"    \", key, \":\", str(value), end=' ')\n",
    "        print()\n",
    "    print(\"SSBD searh results:\")\n",
    "    for j in ssbd_result:\n",
    "        for key, value in j.items():\n",
    "            print(\"    \", key, \":\", str(value), end=' ')\n",
    "        print()\n",
    "    return({'idr': idr_result, 'ssbd': ssbd_result})\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Searching for the term \"mus\" in both SSBD-OMERO and IDR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "IDR searh results:\n",
      "     id : 502      Organism : Mus musculus \n",
      "     id : 405      Organism : Mus musculus x Mus spretus \n",
      "     id : 353      Organism : Mus musculus \n",
      "     id : 352      Organism : Mus musculus \n",
      "     id : 351      Organism : Mus musculus \n",
      "     id : 101      Organism : Mus musculus \n",
      "     id : 301      Organism : Mus musculus \n",
      "SSBD searh results:\n",
      "     id : 59      Organism : Mus musculus \n",
      "     id : 9      Organism : M. musculus \n",
      "     id : 106      Organism : Mus musculus \n",
      "     id : 105      Organism : Mus musculus \n",
      "     id : 104      Organism : Mus musculus \n",
      "     id : 70      Organism : Mus musculus \n",
      "     id : 12      Organism : M. musculus \n",
      "     id : 3      Organism : M. musculus \n",
      "     id : 55      Organism : Mus musculus \n",
      "     id : 103      Organism : Mus musculus \n",
      "     id : 56      Organism : Mus musculus \n",
      "     id : 57      Organism : Mus musculus \n",
      "     id : 58      Organism : Mus musculus \n",
      "     id : 7      Organism : M. musculus \n",
      "     id : 152      Organism : M. musculus \n",
      "     id : 154      Organism : M. musculus \n",
      "     id : 159      Organism : M. musculus \n",
      "     id : 171      Organism : M. musculus \n",
      "     id : 168      Organism : M. musculus \n",
      "     id : 169      Organism : M. musculus \n",
      "     id : 151      Organism : M. musculus \n",
      "     id : 173      Organism : M. musculus \n",
      "     id : 153      Organism : M. musculus \n",
      "     id : 155      Organism : M. musculus \n",
      "     id : 160      Organism : M. musculus \n",
      "     id : 162      Organism : M. musculus \n",
      "     id : 172      Organism : M. musculus \n",
      "     id : 201      Organism : M. musculus \n"
     ]
    }
   ],
   "source": [
    "all_result = combine_search('mus')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'idr': [{'id': 502, 'Organism': 'Mus musculus'},\n",
       "  {'id': 405, 'Organism': 'Mus musculus x Mus spretus'},\n",
       "  {'id': 353, 'Organism': 'Mus musculus'},\n",
       "  {'id': 352, 'Organism': 'Mus musculus'},\n",
       "  {'id': 351, 'Organism': 'Mus musculus'},\n",
       "  {'id': 101, 'Organism': 'Mus musculus'},\n",
       "  {'id': 301, 'Organism': 'Mus musculus'}],\n",
       " 'ssbd': [{'id': 59, 'Organism': 'Mus musculus'},\n",
       "  {'id': 9, 'Organism': 'M. musculus'},\n",
       "  {'id': 106, 'Organism': 'Mus musculus'},\n",
       "  {'id': 105, 'Organism': 'Mus musculus'},\n",
       "  {'id': 104, 'Organism': 'Mus musculus'},\n",
       "  {'id': 70, 'Organism': 'Mus musculus'},\n",
       "  {'id': 12, 'Organism': 'M. musculus'},\n",
       "  {'id': 3, 'Organism': 'M. musculus'},\n",
       "  {'id': 55, 'Organism': 'Mus musculus'},\n",
       "  {'id': 103, 'Organism': 'Mus musculus'},\n",
       "  {'id': 56, 'Organism': 'Mus musculus'},\n",
       "  {'id': 57, 'Organism': 'Mus musculus'},\n",
       "  {'id': 58, 'Organism': 'Mus musculus'},\n",
       "  {'id': 7, 'Organism': 'M. musculus'},\n",
       "  {'id': 152, 'Organism': 'M. musculus'},\n",
       "  {'id': 154, 'Organism': 'M. musculus'},\n",
       "  {'id': 159, 'Organism': 'M. musculus'},\n",
       "  {'id': 171, 'Organism': 'M. musculus'},\n",
       "  {'id': 168, 'Organism': 'M. musculus'},\n",
       "  {'id': 169, 'Organism': 'M. musculus'},\n",
       "  {'id': 151, 'Organism': 'M. musculus'},\n",
       "  {'id': 173, 'Organism': 'M. musculus'},\n",
       "  {'id': 153, 'Organism': 'M. musculus'},\n",
       "  {'id': 155, 'Organism': 'M. musculus'},\n",
       "  {'id': 160, 'Organism': 'M. musculus'},\n",
       "  {'id': 162, 'Organism': 'M. musculus'},\n",
       "  {'id': 172, 'Organism': 'M. musculus'},\n",
       "  {'id': 201, 'Organism': 'M. musculus'}]}"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
